// vue-tailwind-datepicker/dist/vue-tailwind-datepicker.js
(function(){"use strict";try{if(typeof document<"u"){var r=document.createElement("style");r.appendChild(document.createTextNode('.vtd-datepicker-overlay.open:before{display:block;opacity:.5}.vtd-datepicker:before{--vtd-datepicker: 0px;content:"";position:absolute;top:0;height:1rem;width:1rem;border-width:1px;border-color:#0000001a;--tw-bg-opacity: 1;background-color:rgb(255 255 255 / var(--tw-bg-opacity));--tw-shadow: 0 1px 3px 0 rgb(0 0 0 / .1), 0 1px 2px -1px rgb(0 0 0 / .1);--tw-shadow-colored: 0 1px 3px 0 var(--tw-shadow-color), 0 1px 2px -1px var(--tw-shadow-color);box-shadow:var(--tw-ring-offset-shadow, 0 0 #0000),var(--tw-ring-shadow, 0 0 #0000),var(--tw-shadow)}:is(.dark .vtd-datepicker):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}.vtd-datepicker:before{transform:translate(50%,-50%) rotate(-45deg);-webkit-clip-path:polygon(calc(var(--vtd-datepicker) * -1) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(100% + var(--vtd-datepicker)));clip-path:polygon(calc(var(--vtd-datepicker) * -1) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(var(--vtd-datepicker) * -1),calc(100% + var(--vtd-datepicker)) calc(100% + var(--vtd-datepicker)))}.vtd-datepicker.place-left:before{left:.25rem}:is(.dark .vtd-datepicker.place-left):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}.vtd-datepicker.place-right:before{right:1.25rem}:is(.dark .vtd-datepicker.place-right):before{--tw-border-opacity: 1;border-color:rgb(55 65 81 / var(--tw-border-opacity));--tw-bg-opacity: 1;background-color:rgb(31 41 55 / var(--tw-bg-opacity))}')),document.head.appendChild(r)}}catch(t){console.error("vite-plugin-css-injected-by-js",t)}})();
import { cloneVNode as Zt, h as me, Fragment as xe, inject as Oe, provide as se, ref as q, onMounted as rt, watchEffect as Me, computed as ue, defineComponent as ie, onUnmounted as et, Teleport as Jt, reactive as pt, shallowRef as qt, openBlock as W, createElementBlock as Z, createElementVNode as C, withDirectives as ve, vShow as pe, toDisplayString as ae, renderList as ze, withModifiers as Se, unref as le, createCommentVNode as ye, createVNode as ee, TransitionGroup as Xt, withCtx as Fe, normalizeClass as Ye, Transition as mt, nextTick as He, watch as Qt, isProxy as er, createBlock as Ue, renderSlot as dt, mergeProps as tr, vModelText as rr } from "vue";
function Pe(t, n, ...u) {
  if (t in n) {
    let a = n[t];
    return typeof a == "function" ? a(...u) : a;
  }
  let e = new Error(`Tried to handle "${t}" but there is no handler defined. Only defined handlers are: ${Object.keys(n).map((a) => `"${a}"`).join(", ")}.`);
  throw Error.captureStackTrace && Error.captureStackTrace(e, Pe), e;
}
var We = ((t) => (t[t.None = 0] = "None", t[t.RenderStrategy = 1] = "RenderStrategy", t[t.Static = 2] = "Static", t))(We || {}), nr = ((t) => (t[t.Unmount = 0] = "Unmount", t[t.Hidden = 1] = "Hidden", t))(nr || {});
function Ve({ visible: t = !0, features: n = 0, ourProps: u, theirProps: e, ...a }) {
  var o;
  let m = yt(e, u), s = Object.assign(a, { props: m });
  if (t || n & 2 && m.static)
    return st(s);
  if (n & 1) {
    let h = (o = m.unmount) == null || o ? 0 : 1;
    return Pe(h, { 0() {
      return null;
    }, 1() {
      return st({ ...a, props: { ...m, hidden: !0, style: { display: "none" } } });
    } });
  }
  return st(s);
}
function st({ props: t, attrs: n, slots: u, slot: e, name: a }) {
  var o, m;
  let { as: s, ...h } = ar(t, ["unmount", "static"]), c = (o = u.default) == null ? void 0 : o.call(u, e), g = {};
  if (e) {
    let b = !1, D = [];
    for (let [T, I] of Object.entries(e))
      typeof I == "boolean" && (b = !0), I === !0 && D.push(T);
    b && (g["data-headlessui-state"] = D.join(" "));
  }
  if (s === "template") {
    if (c = ht(c ?? []), Object.keys(h).length > 0 || Object.keys(n).length > 0) {
      let [b, ...D] = c ?? [];
      if (!or(b) || D.length > 0)
        throw new Error(['Passing props on "template"!', "", `The current component <${a} /> is rendering a "template".`, "However we need to passthrough the following props:", Object.keys(h).concat(Object.keys(n)).map((x) => x.trim()).filter((x, O, E) => E.indexOf(x) === O).sort((x, O) => x.localeCompare(O)).map((x) => `  - ${x}`).join(`
`), "", "You can apply a few solutions:", ['Add an `as="..."` prop, to ensure that we render an actual element instead of a "template".', "Render a single element as the child so that we can forward the props onto that element."].map((x) => `  - ${x}`).join(`
`)].join(`
`));
      let T = yt((m = b.props) != null ? m : {}, h), I = Zt(b, T);
      for (let x in T)
        x.startsWith("on") && (I.props || (I.props = {}), I.props[x] = T[x]);
      return I;
    }
    return Array.isArray(c) && c.length === 1 ? c[0] : c;
  }
  return me(s, Object.assign({}, h, g), { default: () => c });
}
function ht(t) {
  return t.flatMap((n) => n.type === xe ? ht(n.children) : [n]);
}
function yt(...t) {
  if (t.length === 0)
    return {};
  if (t.length === 1)
    return t[0];
  let n = {}, u = {};
  for (let e of t)
    for (let a in e)
      a.startsWith("on") && typeof e[a] == "function" ? (u[a] != null || (u[a] = []), u[a].push(e[a])) : n[a] = e[a];
  if (n.disabled || n["aria-disabled"])
    return Object.assign(n, Object.fromEntries(Object.keys(u).map((e) => [e, void 0])));
  for (let e in u)
    Object.assign(n, { [e](a, ...o) {
      let m = u[e];
      for (let s of m) {
        if (a instanceof Event && a.defaultPrevented)
          return;
        s(a, ...o);
      }
    } });
  return n;
}
function ar(t, n = []) {
  let u = Object.assign({}, t);
  for (let e of n)
    e in u && delete u[e];
  return u;
}
function or(t) {
  return t == null ? !1 : typeof t.type == "string" || typeof t.type == "object" || typeof t.type == "function";
}
let sr = 0;
function lr() {
  return ++sr;
}
function Be() {
  return lr();
}
var Ae = ((t) => (t.Space = " ", t.Enter = "Enter", t.Escape = "Escape", t.Backspace = "Backspace", t.Delete = "Delete", t.ArrowLeft = "ArrowLeft", t.ArrowUp = "ArrowUp", t.ArrowRight = "ArrowRight", t.ArrowDown = "ArrowDown", t.Home = "Home", t.End = "End", t.PageUp = "PageUp", t.PageDown = "PageDown", t.Tab = "Tab", t))(Ae || {});
function H(t) {
  var n;
  return t == null || t.value == null ? null : (n = t.value.$el) != null ? n : t.value;
}
let bt = Symbol("Context");
var Le = ((t) => (t[t.Open = 1] = "Open", t[t.Closed = 2] = "Closed", t[t.Closing = 4] = "Closing", t[t.Opening = 8] = "Opening", t))(Le || {});
function gt() {
  return Oe(bt, null);
}
function ur(t) {
  se(bt, t);
}
function ct(t, n) {
  if (t)
    return t;
  let u = n ?? "button";
  if (typeof u == "string" && u.toLowerCase() === "button")
    return "button";
}
function ir(t, n) {
  let u = q(ct(t.value.type, t.value.as));
  return rt(() => {
    u.value = ct(t.value.type, t.value.as);
  }), Me(() => {
    var e;
    u.value || H(n) && H(n) instanceof HTMLButtonElement && !((e = H(n)) != null && e.hasAttribute("type")) && (u.value = "button");
  }), u;
}
var dr = Object.defineProperty, cr = (t, n, u) => n in t ? dr(t, n, { enumerable: !0, configurable: !0, writable: !0, value: u }) : t[n] = u, ft = (t, n, u) => (cr(t, typeof n != "symbol" ? n + "" : n, u), u);
class fr {
  constructor() {
    ft(this, "current", this.detect()), ft(this, "currentId", 0);
  }
  set(n) {
    this.current !== n && (this.currentId = 0, this.current = n);
  }
  reset() {
    this.set(this.detect());
  }
  nextId() {
    return ++this.currentId;
  }
  get isServer() {
    return this.current === "server";
  }
  get isClient() {
    return this.current === "client";
  }
  detect() {
    return typeof window > "u" || typeof document > "u" ? "server" : "client";
  }
}
let nt = new fr();
function Ce(t) {
  if (nt.isServer)
    return null;
  if (t instanceof Node)
    return t.ownerDocument;
  if (t != null && t.hasOwnProperty("value")) {
    let n = H(t);
    if (n)
      return n.ownerDocument;
  }
  return document;
}
let lt = ["[contentEditable=true]", "[tabindex]", "a[href]", "area[href]", "button:not([disabled])", "iframe", "input:not([disabled])", "select:not([disabled])", "textarea:not([disabled])"].map((t) => `${t}:not([tabindex='-1'])`).join(",");
var _e = ((t) => (t[t.First = 1] = "First", t[t.Previous = 2] = "Previous", t[t.Next = 4] = "Next", t[t.Last = 8] = "Last", t[t.WrapAround = 16] = "WrapAround", t[t.NoScroll = 32] = "NoScroll", t))(_e || {}), tt = ((t) => (t[t.Error = 0] = "Error", t[t.Overflow = 1] = "Overflow", t[t.Success = 2] = "Success", t[t.Underflow = 3] = "Underflow", t))(tt || {}), vr = ((t) => (t[t.Previous = -1] = "Previous", t[t.Next = 1] = "Next", t))(vr || {});
function at(t = document.body) {
  return t == null ? [] : Array.from(t.querySelectorAll(lt)).sort((n, u) => Math.sign((n.tabIndex || Number.MAX_SAFE_INTEGER) - (u.tabIndex || Number.MAX_SAFE_INTEGER)));
}
var it = ((t) => (t[t.Strict = 0] = "Strict", t[t.Loose = 1] = "Loose", t))(it || {});
function xt(t, n = 0) {
  var u;
  return t === ((u = Ce(t)) == null ? void 0 : u.body) ? !1 : Pe(n, { 0() {
    return t.matches(lt);
  }, 1() {
    let e = t;
    for (; e !== null; ) {
      if (e.matches(lt))
        return !0;
      e = e.parentElement;
    }
    return !1;
  } });
}
var pr = ((t) => (t[t.Keyboard = 0] = "Keyboard", t[t.Mouse = 1] = "Mouse", t))(pr || {});
typeof window < "u" && typeof document < "u" && (document.addEventListener("keydown", (t) => {
  t.metaKey || t.altKey || t.ctrlKey || (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0), document.addEventListener("click", (t) => {
  t.detail === 1 ? delete document.documentElement.dataset.headlessuiFocusVisible : t.detail === 0 && (document.documentElement.dataset.headlessuiFocusVisible = "");
}, !0));
let mr = ["textarea", "input"].join(",");
function hr(t) {
  var n, u;
  return (u = (n = t == null ? void 0 : t.matches) == null ? void 0 : n.call(t, mr)) != null ? u : !1;
}
function yr(t, n = (u) => u) {
  return t.slice().sort((u, e) => {
    let a = n(u), o = n(e);
    if (a === null || o === null)
      return 0;
    let m = a.compareDocumentPosition(o);
    return m & Node.DOCUMENT_POSITION_FOLLOWING ? -1 : m & Node.DOCUMENT_POSITION_PRECEDING ? 1 : 0;
  });
}
function Ee(t, n, { sorted: u = !0, relativeTo: e = null, skipElements: a = [] } = {}) {
  var o;
  let m = (o = Array.isArray(t) ? t.length > 0 ? t[0].ownerDocument : document : t == null ? void 0 : t.ownerDocument) != null ? o : document, s = Array.isArray(t) ? u ? yr(t) : t : at(t);
  a.length > 0 && s.length > 1 && (s = s.filter((I) => !a.includes(I))), e = e ?? m.activeElement;
  let h = (() => {
    if (n & 5)
      return 1;
    if (n & 10)
      return -1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), c = (() => {
    if (n & 1)
      return 0;
    if (n & 2)
      return Math.max(0, s.indexOf(e)) - 1;
    if (n & 4)
      return Math.max(0, s.indexOf(e)) + 1;
    if (n & 8)
      return s.length - 1;
    throw new Error("Missing Focus.First, Focus.Previous, Focus.Next or Focus.Last");
  })(), g = n & 32 ? { preventScroll: !0 } : {}, b = 0, D = s.length, T;
  do {
    if (b >= D || b + D <= 0)
      return 0;
    let I = c + b;
    if (n & 16)
      I = (I + D) % D;
    else {
      if (I < 0)
        return 3;
      if (I >= D)
        return 1;
    }
    T = s[I], T == null || T.focus(g), b += h;
  } while (T !== m.activeElement);
  return n & 6 && hr(T) && T.select(), 2;
}
function Ge(t, n, u) {
  nt.isServer || Me((e) => {
    document.addEventListener(t, n, u), e(() => document.removeEventListener(t, n, u));
  });
}
function wt(t, n, u) {
  nt.isServer || Me((e) => {
    window.addEventListener(t, n, u), e(() => window.removeEventListener(t, n, u));
  });
}
function br(t, n, u = ue(() => !0)) {
  function e(o, m) {
    if (!u.value || o.defaultPrevented)
      return;
    let s = m(o);
    if (s === null || !s.getRootNode().contains(s))
      return;
    let h = function c(g) {
      return typeof g == "function" ? c(g()) : Array.isArray(g) || g instanceof Set ? g : [g];
    }(t);
    for (let c of h) {
      if (c === null)
        continue;
      let g = c instanceof HTMLElement ? c : H(c);
      if (g != null && g.contains(s) || o.composed && o.composedPath().includes(g))
        return;
    }
    return !xt(s, it.Loose) && s.tabIndex !== -1 && o.preventDefault(), n(o, s);
  }
  let a = q(null);
  Ge("pointerdown", (o) => {
    var m, s;
    u.value && (a.value = ((s = (m = o.composedPath) == null ? void 0 : m.call(o)) == null ? void 0 : s[0]) || o.target);
  }, !0), Ge("mousedown", (o) => {
    var m, s;
    u.value && (a.value = ((s = (m = o.composedPath) == null ? void 0 : m.call(o)) == null ? void 0 : s[0]) || o.target);
  }, !0), Ge("click", (o) => {
    a.value && (e(o, () => a.value), a.value = null);
  }, !0), Ge("touchend", (o) => e(o, () => o.target instanceof HTMLElement ? o.target : null), !0), wt("blur", (o) => e(o, () => window.document.activeElement instanceof HTMLIFrameElement ? window.document.activeElement : null), !0);
}
var Ne = ((t) => (t[t.None = 1] = "None", t[t.Focusable = 2] = "Focusable", t[t.Hidden = 4] = "Hidden", t))(Ne || {});
let Ke = ie({ name: "Hidden", props: { as: { type: [Object, String], default: "div" }, features: { type: Number, default: 1 } }, setup(t, { slots: n, attrs: u }) {
  return () => {
    let { features: e, ...a } = t, o = { "aria-hidden": (e & 2) === 2 ? !0 : void 0, style: { position: "fixed", top: 1, left: 1, width: 1, height: 0, padding: 0, margin: -1, overflow: "hidden", clip: "rect(0, 0, 0, 0)", whiteSpace: "nowrap", borderWidth: "0", ...(e & 4) === 4 && (e & 2) !== 2 && { display: "none" } } };
    return Ve({ ourProps: o, theirProps: a, slot: {}, attrs: u, slots: n, name: "Hidden" });
  };
} });
var De = ((t) => (t[t.Forwards = 0] = "Forwards", t[t.Backwards = 1] = "Backwards", t))(De || {});
function kt() {
  let t = q(0);
  return wt("keydown", (n) => {
    n.key === "Tab" && (t.value = n.shiftKey ? 1 : 0);
  }), t;
}
function gr(t, n, u, e) {
  nt.isServer || Me((a) => {
    t = t ?? window, t.addEventListener(n, u, e), a(() => t.removeEventListener(n, u, e));
  });
}
let $t = Symbol("ForcePortalRootContext");
function xr() {
  return Oe($t, !1);
}
ie({ name: "ForcePortalRoot", props: { as: { type: [Object, String], default: "template" }, force: { type: Boolean, default: !1 } }, setup(t, { slots: n, attrs: u }) {
  return se($t, t.force), () => {
    let { force: e, ...a } = t;
    return Ve({ theirProps: a, ourProps: {}, slot: {}, slots: n, attrs: u, name: "ForcePortalRoot" });
  };
} });
function wr(t) {
  let n = Ce(t);
  if (!n) {
    if (t === null)
      return null;
    throw new Error(`[Headless UI]: Cannot find ownerDocument for contextElement: ${t}`);
  }
  let u = n.getElementById("headlessui-portal-root");
  if (u)
    return u;
  let e = n.createElement("div");
  return e.setAttribute("id", "headlessui-portal-root"), n.body.appendChild(e);
}
ie({ name: "Portal", props: { as: { type: [Object, String], default: "div" } }, setup(t, { slots: n, attrs: u }) {
  let e = q(null), a = ue(() => Ce(e)), o = xr(), m = Oe(Mt, null), s = q(o === !0 || m == null ? wr(e.value) : m.resolveTarget());
  Me(() => {
    o || m != null && (s.value = m.resolveTarget());
  });
  let h = Oe(ut, null);
  return rt(() => {
    let c = H(e);
    c && h && et(h.register(c));
  }), et(() => {
    var c, g;
    let b = (c = a.value) == null ? void 0 : c.getElementById("headlessui-portal-root");
    b && s.value === b && s.value.children.length <= 0 && ((g = s.value.parentElement) == null || g.removeChild(s.value));
  }), () => {
    if (s.value === null)
      return null;
    let c = { ref: e, "data-headlessui-portal": "" };
    return me(Jt, { to: s.value }, Ve({ ourProps: c, theirProps: t, slot: {}, attrs: u, slots: n, name: "Portal" }));
  };
} });
let ut = Symbol("PortalParentContext");
function kr() {
  let t = Oe(ut, null), n = q([]);
  function u(o) {
    return n.value.push(o), t && t.register(o), () => e(o);
  }
  function e(o) {
    let m = n.value.indexOf(o);
    m !== -1 && n.value.splice(m, 1), t && t.unregister(o);
  }
  let a = { register: u, unregister: e, portals: n };
  return [n, ie({ name: "PortalWrapper", setup(o, { slots: m }) {
    return se(ut, a), () => {
      var s;
      return (s = m.default) == null ? void 0 : s.call(m);
    };
  } })];
}
let Mt = Symbol("PortalGroupContext");
ie({ name: "PortalGroup", props: { as: { type: [Object, String], default: "template" }, target: { type: Object, default: null } }, setup(t, { attrs: n, slots: u }) {
  let e = pt({ resolveTarget() {
    return t.target;
  } });
  return se(Mt, e), () => {
    let { target: a, ...o } = t;
    return Ve({ theirProps: o, ourProps: {}, slot: {}, attrs: n, slots: u, name: "PortalGroup" });
  };
} });
function $r({ defaultContainers: t = [], portals: n, mainTreeNodeRef: u } = {}) {
  let e = q(null), a = Ce(e);
  function o() {
    var m;
    let s = [];
    for (let h of t)
      h !== null && (h instanceof HTMLElement ? s.push(h) : "value" in h && h.value instanceof HTMLElement && s.push(h.value));
    if (n != null && n.value)
      for (let h of n.value)
        s.push(h);
    for (let h of (m = a == null ? void 0 : a.querySelectorAll("html > *, body > *")) != null ? m : [])
      h !== document.body && h !== document.head && h instanceof HTMLElement && h.id !== "headlessui-portal-root" && (h.contains(H(e)) || s.some((c) => h.contains(c)) || s.push(h));
    return s;
  }
  return { resolveContainers: o, contains(m) {
    return o().some((s) => s.contains(m));
  }, mainTreeNodeRef: e, MainTreeNode() {
    return u != null ? null : me(Ke, { features: Ne.Hidden, ref: e });
  } };
}
function Mr() {
  let t = q(null);
  return { mainTreeNodeRef: t, MainTreeNode() {
    return me(Ke, { features: Ne.Hidden, ref: t });
  } };
}
var Sr = ((t) => (t[t.Open = 0] = "Open", t[t.Closed = 1] = "Closed", t))(Sr || {});
let St = Symbol("PopoverContext");
function ot(t) {
  let n = Oe(St, null);
  if (n === null) {
    let u = new Error(`<${t} /> is missing a parent <${Pt.name} /> component.`);
    throw Error.captureStackTrace && Error.captureStackTrace(u, ot), u;
  }
  return n;
}
let _t = Symbol("PopoverGroupContext");
function Dt() {
  return Oe(_t, null);
}
let Ot = Symbol("PopoverPanelContext");
function _r() {
  return Oe(Ot, null);
}
let Pt = ie({ name: "Popover", inheritAttrs: !1, props: { as: { type: [Object, String], default: "div" } }, setup(t, { slots: n, attrs: u, expose: e }) {
  var a;
  let o = q(null);
  e({ el: o, $el: o });
  let m = q(1), s = q(null), h = q(null), c = q(null), g = q(null), b = ue(() => Ce(o)), D = ue(() => {
    var A, P;
    if (!H(s) || !H(g))
      return !1;
    for (let J of document.querySelectorAll("body > *"))
      if (Number(J == null ? void 0 : J.contains(H(s))) ^ Number(J == null ? void 0 : J.contains(H(g))))
        return !0;
    let l = at(), r = l.indexOf(H(s)), R = (r + l.length - 1) % l.length, N = (r + 1) % l.length, M = l[R], z = l[N];
    return !((A = H(g)) != null && A.contains(M)) && !((P = H(g)) != null && P.contains(z));
  }), T = { popoverState: m, buttonId: q(null), panelId: q(null), panel: g, button: s, isPortalled: D, beforePanelSentinel: h, afterPanelSentinel: c, togglePopover() {
    m.value = Pe(m.value, { 0: 1, 1: 0 });
  }, closePopover() {
    m.value !== 1 && (m.value = 1);
  }, close(A) {
    T.closePopover();
    let P = (() => A ? A instanceof HTMLElement ? A : A.value instanceof HTMLElement ? H(A) : H(T.button) : H(T.button))();
    P == null || P.focus();
  } };
  se(St, T), ur(ue(() => Pe(m.value, { 0: Le.Open, 1: Le.Closed })));
  let I = { buttonId: T.buttonId, panelId: T.panelId, close() {
    T.closePopover();
  } }, x = Dt(), O = x == null ? void 0 : x.registerPopover, [E, k] = kr(), w = $r({ mainTreeNodeRef: x == null ? void 0 : x.mainTreeNodeRef, portals: E, defaultContainers: [s, g] });
  function V() {
    var A, P, l, r;
    return (r = x == null ? void 0 : x.isFocusWithinPopoverGroup()) != null ? r : ((A = b.value) == null ? void 0 : A.activeElement) && (((P = H(s)) == null ? void 0 : P.contains(b.value.activeElement)) || ((l = H(g)) == null ? void 0 : l.contains(b.value.activeElement)));
  }
  return Me(() => O == null ? void 0 : O(I)), gr((a = b.value) == null ? void 0 : a.defaultView, "focus", (A) => {
    var P, l;
    A.target !== window && A.target instanceof HTMLElement && m.value === 0 && (V() || s && g && (w.contains(A.target) || (P = H(T.beforePanelSentinel)) != null && P.contains(A.target) || (l = H(T.afterPanelSentinel)) != null && l.contains(A.target) || T.closePopover()));
  }, !0), br(w.resolveContainers, (A, P) => {
    var l;
    T.closePopover(), xt(P, it.Loose) || (A.preventDefault(), (l = H(s)) == null || l.focus());
  }, ue(() => m.value === 0)), () => {
    let A = { open: m.value === 0, close: T.close };
    return me(xe, [me(k, {}, () => Ve({ theirProps: { ...t, ...u }, ourProps: { ref: o }, slot: A, slots: n, attrs: u, name: "Popover" })), me(w.MainTreeNode)]);
  };
} }), Dr = ie({ name: "PopoverButton", props: { as: { type: [Object, String], default: "button" }, disabled: { type: [Boolean], default: !1 }, id: { type: String, default: () => `headlessui-popover-button-${Be()}` } }, inheritAttrs: !1, setup(t, { attrs: n, slots: u, expose: e }) {
  let a = ot("PopoverButton"), o = ue(() => Ce(a.button));
  e({ el: a.button, $el: a.button }), rt(() => {
    a.buttonId.value = t.id;
  }), et(() => {
    a.buttonId.value = null;
  });
  let m = Dt(), s = m == null ? void 0 : m.closeOthers, h = _r(), c = ue(() => h === null ? !1 : h.value === a.panelId.value), g = q(null), b = `headlessui-focus-sentinel-${Be()}`;
  c.value || Me(() => {
    a.button.value = g.value;
  });
  let D = ir(ue(() => ({ as: t.as, type: n.type })), g);
  function T(w) {
    var V, A, P, l, r;
    if (c.value) {
      if (a.popoverState.value === 1)
        return;
      switch (w.key) {
        case Ae.Space:
        case Ae.Enter:
          w.preventDefault(), (A = (V = w.target).click) == null || A.call(V), a.closePopover(), (P = H(a.button)) == null || P.focus();
          break;
      }
    } else
      switch (w.key) {
        case Ae.Space:
        case Ae.Enter:
          w.preventDefault(), w.stopPropagation(), a.popoverState.value === 1 && (s == null || s(a.buttonId.value)), a.togglePopover();
          break;
        case Ae.Escape:
          if (a.popoverState.value !== 0)
            return s == null ? void 0 : s(a.buttonId.value);
          if (!H(a.button) || (l = o.value) != null && l.activeElement && !((r = H(a.button)) != null && r.contains(o.value.activeElement)))
            return;
          w.preventDefault(), w.stopPropagation(), a.closePopover();
          break;
      }
  }
  function I(w) {
    c.value || w.key === Ae.Space && w.preventDefault();
  }
  function x(w) {
    var V, A;
    t.disabled || (c.value ? (a.closePopover(), (V = H(a.button)) == null || V.focus()) : (w.preventDefault(), w.stopPropagation(), a.popoverState.value === 1 && (s == null || s(a.buttonId.value)), a.togglePopover(), (A = H(a.button)) == null || A.focus()));
  }
  function O(w) {
    w.preventDefault(), w.stopPropagation();
  }
  let E = kt();
  function k() {
    let w = H(a.panel);
    if (!w)
      return;
    function V() {
      Pe(E.value, { [De.Forwards]: () => Ee(w, _e.First), [De.Backwards]: () => Ee(w, _e.Last) }) === tt.Error && Ee(at().filter((A) => A.dataset.headlessuiFocusGuard !== "true"), Pe(E.value, { [De.Forwards]: _e.Next, [De.Backwards]: _e.Previous }), { relativeTo: H(a.button) });
    }
    V();
  }
  return () => {
    let w = a.popoverState.value === 0, V = { open: w }, { id: A, ...P } = t, l = c.value ? { ref: g, type: D.value, onKeydown: T, onClick: x } : { ref: g, id: A, type: D.value, "aria-expanded": a.popoverState.value === 0, "aria-controls": H(a.panel) ? a.panelId.value : void 0, disabled: t.disabled ? !0 : void 0, onKeydown: T, onKeyup: I, onClick: x, onMousedown: O };
    return me(xe, [Ve({ ourProps: l, theirProps: { ...n, ...P }, slot: V, attrs: n, slots: u, name: "PopoverButton" }), w && !c.value && a.isPortalled.value && me(Ke, { id: b, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: k })]);
  };
} }), Or = ie({ name: "PopoverOverlay", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 } }, setup(t, { attrs: n, slots: u }) {
  let e = ot("PopoverOverlay"), a = `headlessui-popover-overlay-${Be()}`, o = gt(), m = ue(() => o !== null ? (o.value & Le.Open) === Le.Open : e.popoverState.value === 0);
  function s() {
    e.closePopover();
  }
  return () => {
    let h = { open: e.popoverState.value === 0 };
    return Ve({ ourProps: { id: a, "aria-hidden": !0, onClick: s }, theirProps: t, slot: h, attrs: n, slots: u, features: We.RenderStrategy | We.Static, visible: m.value, name: "PopoverOverlay" });
  };
} }), Pr = ie({ name: "PopoverPanel", props: { as: { type: [Object, String], default: "div" }, static: { type: Boolean, default: !1 }, unmount: { type: Boolean, default: !0 }, focus: { type: Boolean, default: !1 }, id: { type: String, default: () => `headlessui-popover-panel-${Be()}` } }, inheritAttrs: !1, setup(t, { attrs: n, slots: u, expose: e }) {
  let { focus: a } = t, o = ot("PopoverPanel"), m = ue(() => Ce(o.panel)), s = `headlessui-focus-sentinel-before-${Be()}`, h = `headlessui-focus-sentinel-after-${Be()}`;
  e({ el: o.panel, $el: o.panel }), rt(() => {
    o.panelId.value = t.id;
  }), et(() => {
    o.panelId.value = null;
  }), se(Ot, o.panelId), Me(() => {
    var O, E;
    if (!a || o.popoverState.value !== 0 || !o.panel)
      return;
    let k = (O = m.value) == null ? void 0 : O.activeElement;
    (E = H(o.panel)) != null && E.contains(k) || Ee(H(o.panel), _e.First);
  });
  let c = gt(), g = ue(() => c !== null ? (c.value & Le.Open) === Le.Open : o.popoverState.value === 0);
  function b(O) {
    var E, k;
    switch (O.key) {
      case Ae.Escape:
        if (o.popoverState.value !== 0 || !H(o.panel) || m.value && !((E = H(o.panel)) != null && E.contains(m.value.activeElement)))
          return;
        O.preventDefault(), O.stopPropagation(), o.closePopover(), (k = H(o.button)) == null || k.focus();
        break;
    }
  }
  function D(O) {
    var E, k, w, V, A;
    let P = O.relatedTarget;
    P && H(o.panel) && ((E = H(o.panel)) != null && E.contains(P) || (o.closePopover(), ((w = (k = H(o.beforePanelSentinel)) == null ? void 0 : k.contains) != null && w.call(k, P) || (A = (V = H(o.afterPanelSentinel)) == null ? void 0 : V.contains) != null && A.call(V, P)) && P.focus({ preventScroll: !0 })));
  }
  let T = kt();
  function I() {
    let O = H(o.panel);
    if (!O)
      return;
    function E() {
      Pe(T.value, { [De.Forwards]: () => {
        var k;
        Ee(O, _e.First) === tt.Error && ((k = H(o.afterPanelSentinel)) == null || k.focus());
      }, [De.Backwards]: () => {
        var k;
        (k = H(o.button)) == null || k.focus({ preventScroll: !0 });
      } });
    }
    E();
  }
  function x() {
    let O = H(o.panel);
    if (!O)
      return;
    function E() {
      Pe(T.value, { [De.Forwards]: () => {
        let k = H(o.button), w = H(o.panel);
        if (!k)
          return;
        let V = at(), A = V.indexOf(k), P = V.slice(0, A + 1), l = [...V.slice(A + 1), ...P];
        for (let r of l.slice())
          if (r.dataset.headlessuiFocusGuard === "true" || w != null && w.contains(r)) {
            let R = l.indexOf(r);
            R !== -1 && l.splice(R, 1);
          }
        Ee(l, _e.First, { sorted: !1 });
      }, [De.Backwards]: () => {
        var k;
        Ee(O, _e.Previous) === tt.Error && ((k = H(o.button)) == null || k.focus());
      } });
    }
    E();
  }
  return () => {
    let O = { open: o.popoverState.value === 0, close: o.close }, { id: E, focus: k, ...w } = t, V = { ref: o.panel, id: E, onKeydown: b, onFocusout: a && o.popoverState.value === 0 ? D : void 0, tabIndex: -1 };
    return Ve({ ourProps: V, theirProps: { ...n, ...w }, attrs: n, slot: O, slots: { ...u, default: (...A) => {
      var P;
      return [me(xe, [g.value && o.isPortalled.value && me(Ke, { id: s, ref: o.beforePanelSentinel, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: I }), (P = u.default) == null ? void 0 : P.call(u, ...A), g.value && o.isPortalled.value && me(Ke, { id: h, ref: o.afterPanelSentinel, features: Ne.Focusable, "data-headlessui-focus-guard": !0, as: "button", type: "button", onFocus: x })])];
    } }, features: We.RenderStrategy | We.Static, visible: g.value, name: "PopoverPanel" });
  };
} });
ie({ name: "PopoverGroup", inheritAttrs: !1, props: { as: { type: [Object, String], default: "div" } }, setup(t, { attrs: n, slots: u, expose: e }) {
  let a = q(null), o = qt([]), m = ue(() => Ce(a)), s = Mr();
  e({ el: a, $el: a });
  function h(D) {
    let T = o.value.indexOf(D);
    T !== -1 && o.value.splice(T, 1);
  }
  function c(D) {
    return o.value.push(D), () => {
      h(D);
    };
  }
  function g() {
    var D;
    let T = m.value;
    if (!T)
      return !1;
    let I = T.activeElement;
    return (D = H(a)) != null && D.contains(I) ? !0 : o.value.some((x) => {
      var O, E;
      return ((O = T.getElementById(x.buttonId.value)) == null ? void 0 : O.contains(I)) || ((E = T.getElementById(x.panelId.value)) == null ? void 0 : E.contains(I));
    });
  }
  function b(D) {
    for (let T of o.value)
      T.buttonId.value !== D && T.close();
  }
  return se(_t, { registerPopover: c, unregisterPopover: h, isFocusWithinPopoverGroup: g, closeOthers: b, mainTreeNodeRef: s.mainTreeNodeRef }), () => me(xe, [Ve({ ourProps: { ref: a }, theirProps: { ...t, ...n }, slot: {}, attrs: n, slots: u, name: "PopoverGroup" }), me(s.MainTreeNode)]);
} });
var Te = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {};
function je(t) {
  return t && t.__esModule && Object.prototype.hasOwnProperty.call(t, "default") ? t.default : t;
}
var Vt = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    var u = 1e3, e = 6e4, a = 36e5, o = "millisecond", m = "second", s = "minute", h = "hour", c = "day", g = "week", b = "month", D = "quarter", T = "year", I = "date", x = "Invalid Date", O = /^(\d{4})[-/]?(\d{1,2})?[-/]?(\d{0,2})[Tt\s]*(\d{1,2})?:?(\d{1,2})?:?(\d{1,2})?[.:]?(\d+)?$/, E = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, k = { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(j) {
      var y = ["th", "st", "nd", "rd"], _ = j % 100;
      return "[" + j + (y[(_ - 20) % 10] || y[_] || y[0]) + "]";
    } }, w = function(j, y, _) {
      var Y = String(j);
      return !Y || Y.length >= y ? j : "" + Array(y + 1 - Y.length).join(_) + j;
    }, V = { s: w, z: function(j) {
      var y = -j.utcOffset(), _ = Math.abs(y), Y = Math.floor(_ / 60), S = _ % 60;
      return (y <= 0 ? "+" : "-") + w(Y, 2, "0") + ":" + w(S, 2, "0");
    }, m: function j(y, _) {
      if (y.date() < _.date())
        return -j(_, y);
      var Y = 12 * (_.year() - y.year()) + (_.month() - y.month()), S = y.clone().add(Y, b), B = _ - S < 0, F = y.clone().add(Y + (B ? -1 : 1), b);
      return +(-(Y + (_ - S) / (B ? S - F : F - S)) || 0);
    }, a: function(j) {
      return j < 0 ? Math.ceil(j) || 0 : Math.floor(j);
    }, p: function(j) {
      return { M: b, y: T, w: g, d: c, D: I, h, m: s, s: m, ms: o, Q: D }[j] || String(j || "").toLowerCase().replace(/s$/, "");
    }, u: function(j) {
      return j === void 0;
    } }, A = "en", P = {};
    P[A] = k;
    var l = "$isDayjsObject", r = function(j) {
      return j instanceof z || !(!j || !j[l]);
    }, R = function j(y, _, Y) {
      var S;
      if (!y)
        return A;
      if (typeof y == "string") {
        var B = y.toLowerCase();
        P[B] && (S = B), _ && (P[B] = _, S = B);
        var F = y.split("-");
        if (!S && F.length > 1)
          return j(F[0]);
      } else {
        var G = y.name;
        P[G] = y, S = G;
      }
      return !Y && S && (A = S), S || !Y && A;
    }, N = function(j, y) {
      if (r(j))
        return j.clone();
      var _ = typeof y == "object" ? y : {};
      return _.date = j, _.args = arguments, new z(_);
    }, M = V;
    M.l = R, M.i = r, M.w = function(j, y) {
      return N(j, { locale: y.$L, utc: y.$u, x: y.$x, $offset: y.$offset });
    };
    var z = function() {
      function j(_) {
        this.$L = R(_.locale, null, !0), this.parse(_), this.$x = this.$x || _.x || {}, this[l] = !0;
      }
      var y = j.prototype;
      return y.parse = function(_) {
        this.$d = function(Y) {
          var S = Y.date, B = Y.utc;
          if (S === null)
            return /* @__PURE__ */ new Date(NaN);
          if (M.u(S))
            return /* @__PURE__ */ new Date();
          if (S instanceof Date)
            return new Date(S);
          if (typeof S == "string" && !/Z$/i.test(S)) {
            var F = S.match(O);
            if (F) {
              var G = F[2] - 1 || 0, Q = (F[7] || "0").substring(0, 3);
              return B ? new Date(Date.UTC(F[1], G, F[3] || 1, F[4] || 0, F[5] || 0, F[6] || 0, Q)) : new Date(F[1], G, F[3] || 1, F[4] || 0, F[5] || 0, F[6] || 0, Q);
            }
          }
          return new Date(S);
        }(_), this.init();
      }, y.init = function() {
        var _ = this.$d;
        this.$y = _.getFullYear(), this.$M = _.getMonth(), this.$D = _.getDate(), this.$W = _.getDay(), this.$H = _.getHours(), this.$m = _.getMinutes(), this.$s = _.getSeconds(), this.$ms = _.getMilliseconds();
      }, y.$utils = function() {
        return M;
      }, y.isValid = function() {
        return this.$d.toString() !== x;
      }, y.isSame = function(_, Y) {
        var S = N(_);
        return this.startOf(Y) <= S && S <= this.endOf(Y);
      }, y.isAfter = function(_, Y) {
        return N(_) < this.startOf(Y);
      }, y.isBefore = function(_, Y) {
        return this.endOf(Y) < N(_);
      }, y.$g = function(_, Y, S) {
        return M.u(_) ? this[Y] : this.set(S, _);
      }, y.unix = function() {
        return Math.floor(this.valueOf() / 1e3);
      }, y.valueOf = function() {
        return this.$d.getTime();
      }, y.startOf = function(_, Y) {
        var S = this, B = !!M.u(Y) || Y, F = M.p(_), G = function(he, ne) {
          var fe = M.w(S.$u ? Date.UTC(S.$y, ne, he) : new Date(S.$y, ne, he), S);
          return B ? fe : fe.endOf(c);
        }, Q = function(he, ne) {
          return M.w(S.toDate()[he].apply(S.toDate("s"), (B ? [0, 0, 0, 0] : [23, 59, 59, 999]).slice(ne)), S);
        }, te = this.$W, re = this.$M, de = this.$D, we = "set" + (this.$u ? "UTC" : "");
        switch (F) {
          case T:
            return B ? G(1, 0) : G(31, 11);
          case b:
            return B ? G(1, re) : G(0, re + 1);
          case g:
            var oe = this.$locale().weekStart || 0, be = (te < oe ? te + 7 : te) - oe;
            return G(B ? de - be : de + (6 - be), re);
          case c:
          case I:
            return Q(we + "Hours", 0);
          case h:
            return Q(we + "Minutes", 1);
          case s:
            return Q(we + "Seconds", 2);
          case m:
            return Q(we + "Milliseconds", 3);
          default:
            return this.clone();
        }
      }, y.endOf = function(_) {
        return this.startOf(_, !1);
      }, y.$set = function(_, Y) {
        var S, B = M.p(_), F = "set" + (this.$u ? "UTC" : ""), G = (S = {}, S[c] = F + "Date", S[I] = F + "Date", S[b] = F + "Month", S[T] = F + "FullYear", S[h] = F + "Hours", S[s] = F + "Minutes", S[m] = F + "Seconds", S[o] = F + "Milliseconds", S)[B], Q = B === c ? this.$D + (Y - this.$W) : Y;
        if (B === b || B === T) {
          var te = this.clone().set(I, 1);
          te.$d[G](Q), te.init(), this.$d = te.set(I, Math.min(this.$D, te.daysInMonth())).$d;
        } else
          G && this.$d[G](Q);
        return this.init(), this;
      }, y.set = function(_, Y) {
        return this.clone().$set(_, Y);
      }, y.get = function(_) {
        return this[M.p(_)]();
      }, y.add = function(_, Y) {
        var S, B = this;
        _ = Number(_);
        var F = M.p(Y), G = function(re) {
          var de = N(B);
          return M.w(de.date(de.date() + Math.round(re * _)), B);
        };
        if (F === b)
          return this.set(b, this.$M + _);
        if (F === T)
          return this.set(T, this.$y + _);
        if (F === c)
          return G(1);
        if (F === g)
          return G(7);
        var Q = (S = {}, S[s] = e, S[h] = a, S[m] = u, S)[F] || 1, te = this.$d.getTime() + _ * Q;
        return M.w(te, this);
      }, y.subtract = function(_, Y) {
        return this.add(-1 * _, Y);
      }, y.format = function(_) {
        var Y = this, S = this.$locale();
        if (!this.isValid())
          return S.invalidDate || x;
        var B = _ || "YYYY-MM-DDTHH:mm:ssZ", F = M.z(this), G = this.$H, Q = this.$m, te = this.$M, re = S.weekdays, de = S.months, we = S.meridiem, oe = function(ne, fe, ge, ke) {
          return ne && (ne[fe] || ne(Y, B)) || ge[fe].slice(0, ke);
        }, be = function(ne) {
          return M.s(G % 12 || 12, ne, "0");
        }, he = we || function(ne, fe, ge) {
          var ke = ne < 12 ? "AM" : "PM";
          return ge ? ke.toLowerCase() : ke;
        };
        return B.replace(E, function(ne, fe) {
          return fe || function(ge) {
            switch (ge) {
              case "YY":
                return String(Y.$y).slice(-2);
              case "YYYY":
                return M.s(Y.$y, 4, "0");
              case "M":
                return te + 1;
              case "MM":
                return M.s(te + 1, 2, "0");
              case "MMM":
                return oe(S.monthsShort, te, de, 3);
              case "MMMM":
                return oe(de, te);
              case "D":
                return Y.$D;
              case "DD":
                return M.s(Y.$D, 2, "0");
              case "d":
                return String(Y.$W);
              case "dd":
                return oe(S.weekdaysMin, Y.$W, re, 2);
              case "ddd":
                return oe(S.weekdaysShort, Y.$W, re, 3);
              case "dddd":
                return re[Y.$W];
              case "H":
                return String(G);
              case "HH":
                return M.s(G, 2, "0");
              case "h":
                return be(1);
              case "hh":
                return be(2);
              case "a":
                return he(G, Q, !0);
              case "A":
                return he(G, Q, !1);
              case "m":
                return String(Q);
              case "mm":
                return M.s(Q, 2, "0");
              case "s":
                return String(Y.$s);
              case "ss":
                return M.s(Y.$s, 2, "0");
              case "SSS":
                return M.s(Y.$ms, 3, "0");
              case "Z":
                return F;
            }
            return null;
          }(ne) || F.replace(":", "");
        });
      }, y.utcOffset = function() {
        return 15 * -Math.round(this.$d.getTimezoneOffset() / 15);
      }, y.diff = function(_, Y, S) {
        var B, F = this, G = M.p(Y), Q = N(_), te = (Q.utcOffset() - this.utcOffset()) * e, re = this - Q, de = function() {
          return M.m(F, Q);
        };
        switch (G) {
          case T:
            B = de() / 12;
            break;
          case b:
            B = de();
            break;
          case D:
            B = de() / 3;
            break;
          case g:
            B = (re - te) / 6048e5;
            break;
          case c:
            B = (re - te) / 864e5;
            break;
          case h:
            B = re / a;
            break;
          case s:
            B = re / e;
            break;
          case m:
            B = re / u;
            break;
          default:
            B = re;
        }
        return S ? B : M.a(B);
      }, y.daysInMonth = function() {
        return this.endOf(b).$D;
      }, y.$locale = function() {
        return P[this.$L];
      }, y.locale = function(_, Y) {
        if (!_)
          return this.$L;
        var S = this.clone(), B = R(_, Y, !0);
        return B && (S.$L = B), S;
      }, y.clone = function() {
        return M.w(this.$d, this);
      }, y.toDate = function() {
        return new Date(this.valueOf());
      }, y.toJSON = function() {
        return this.isValid() ? this.toISOString() : null;
      }, y.toISOString = function() {
        return this.$d.toISOString();
      }, y.toString = function() {
        return this.$d.toUTCString();
      }, j;
    }(), J = z.prototype;
    return N.prototype = J, [["$ms", o], ["$s", m], ["$m", s], ["$H", h], ["$W", c], ["$M", b], ["$y", T], ["$D", I]].forEach(function(j) {
      J[j[1]] = function(y) {
        return this.$g(y, j[0], j[1]);
      };
    }), N.extend = function(j, y) {
      return j.$i || (j(y, z, N), j.$i = !0), N;
    }, N.locale = R, N.isDayjs = r, N.unix = function(j) {
      return N(1e3 * j);
    }, N.en = P[A], N.Ls = P, N.p = {}, N;
  });
})(Vt);
var Vr = Vt.exports;
const d = /* @__PURE__ */ je(Vr);
var Tt = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    return function(u, e, a) {
      var o = e.prototype, m = function(b) {
        return b && (b.indexOf ? b : b.s);
      }, s = function(b, D, T, I, x) {
        var O = b.name ? b : b.$locale(), E = m(O[D]), k = m(O[T]), w = E || k.map(function(A) {
          return A.slice(0, I);
        });
        if (!x)
          return w;
        var V = O.weekStart;
        return w.map(function(A, P) {
          return w[(P + (V || 0)) % 7];
        });
      }, h = function() {
        return a.Ls[a.locale()];
      }, c = function(b, D) {
        return b.formats[D] || function(T) {
          return T.replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(I, x, O) {
            return x || O.slice(1);
          });
        }(b.formats[D.toUpperCase()]);
      }, g = function() {
        var b = this;
        return { months: function(D) {
          return D ? D.format("MMMM") : s(b, "months");
        }, monthsShort: function(D) {
          return D ? D.format("MMM") : s(b, "monthsShort", "months", 3);
        }, firstDayOfWeek: function() {
          return b.$locale().weekStart || 0;
        }, weekdays: function(D) {
          return D ? D.format("dddd") : s(b, "weekdays");
        }, weekdaysMin: function(D) {
          return D ? D.format("dd") : s(b, "weekdaysMin", "weekdays", 2);
        }, weekdaysShort: function(D) {
          return D ? D.format("ddd") : s(b, "weekdaysShort", "weekdays", 3);
        }, longDateFormat: function(D) {
          return c(b.$locale(), D);
        }, meridiem: this.$locale().meridiem, ordinal: this.$locale().ordinal };
      };
      o.localeData = function() {
        return g.bind(this)();
      }, a.localeData = function() {
        var b = h();
        return { firstDayOfWeek: function() {
          return b.weekStart || 0;
        }, weekdays: function() {
          return a.weekdays();
        }, weekdaysShort: function() {
          return a.weekdaysShort();
        }, weekdaysMin: function() {
          return a.weekdaysMin();
        }, months: function() {
          return a.months();
        }, monthsShort: function() {
          return a.monthsShort();
        }, longDateFormat: function(D) {
          return c(b, D);
        }, meridiem: b.meridiem, ordinal: b.ordinal };
      }, a.months = function() {
        return s(h(), "months");
      }, a.monthsShort = function() {
        return s(h(), "monthsShort", "months", 3);
      }, a.weekdays = function(b) {
        return s(h(), "weekdays", null, null, b);
      }, a.weekdaysShort = function(b) {
        return s(h(), "weekdaysShort", "weekdays", 3, b);
      }, a.weekdaysMin = function(b) {
        return s(h(), "weekdaysMin", "weekdays", 2, b);
      };
    };
  });
})(Tt);
var Tr = Tt.exports;
const Ar = /* @__PURE__ */ je(Tr);
var At = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    var u = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" };
    return function(e, a, o) {
      var m = a.prototype, s = m.format;
      o.en.formats = u, m.format = function(h) {
        h === void 0 && (h = "YYYY-MM-DDTHH:mm:ssZ");
        var c = this.$locale().formats, g = function(b, D) {
          return b.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(T, I, x) {
            var O = x && x.toUpperCase();
            return I || D[x] || u[x] || D[O].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(E, k, w) {
              return k || w.slice(1);
            });
          });
        }(h, c === void 0 ? {} : c);
        return s.call(this, g);
      };
    };
  });
})(At);
var Yr = At.exports;
const Cr = /* @__PURE__ */ je(Yr);
var Yt = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    var u = { LTS: "h:mm:ss A", LT: "h:mm A", L: "MM/DD/YYYY", LL: "MMMM D, YYYY", LLL: "MMMM D, YYYY h:mm A", LLLL: "dddd, MMMM D, YYYY h:mm A" }, e = /(\[[^[]*\])|([-_:/.,()\s]+)|(A|a|YYYY|YY?|MM?M?M?|Do|DD?|hh?|HH?|mm?|ss?|S{1,3}|z|ZZ?)/g, a = /\d\d/, o = /\d\d?/, m = /\d*[^-_:/,()\s\d]+/, s = {}, h = function(x) {
      return (x = +x) + (x > 68 ? 1900 : 2e3);
    }, c = function(x) {
      return function(O) {
        this[x] = +O;
      };
    }, g = [/[+-]\d\d:?(\d\d)?|Z/, function(x) {
      (this.zone || (this.zone = {})).offset = function(O) {
        if (!O || O === "Z")
          return 0;
        var E = O.match(/([+-]|\d\d)/g), k = 60 * E[1] + (+E[2] || 0);
        return k === 0 ? 0 : E[0] === "+" ? -k : k;
      }(x);
    }], b = function(x) {
      var O = s[x];
      return O && (O.indexOf ? O : O.s.concat(O.f));
    }, D = function(x, O) {
      var E, k = s.meridiem;
      if (k) {
        for (var w = 1; w <= 24; w += 1)
          if (x.indexOf(k(w, 0, O)) > -1) {
            E = w > 12;
            break;
          }
      } else
        E = x === (O ? "pm" : "PM");
      return E;
    }, T = { A: [m, function(x) {
      this.afternoon = D(x, !1);
    }], a: [m, function(x) {
      this.afternoon = D(x, !0);
    }], S: [/\d/, function(x) {
      this.milliseconds = 100 * +x;
    }], SS: [a, function(x) {
      this.milliseconds = 10 * +x;
    }], SSS: [/\d{3}/, function(x) {
      this.milliseconds = +x;
    }], s: [o, c("seconds")], ss: [o, c("seconds")], m: [o, c("minutes")], mm: [o, c("minutes")], H: [o, c("hours")], h: [o, c("hours")], HH: [o, c("hours")], hh: [o, c("hours")], D: [o, c("day")], DD: [a, c("day")], Do: [m, function(x) {
      var O = s.ordinal, E = x.match(/\d+/);
      if (this.day = E[0], O)
        for (var k = 1; k <= 31; k += 1)
          O(k).replace(/\[|\]/g, "") === x && (this.day = k);
    }], M: [o, c("month")], MM: [a, c("month")], MMM: [m, function(x) {
      var O = b("months"), E = (b("monthsShort") || O.map(function(k) {
        return k.slice(0, 3);
      })).indexOf(x) + 1;
      if (E < 1)
        throw new Error();
      this.month = E % 12 || E;
    }], MMMM: [m, function(x) {
      var O = b("months").indexOf(x) + 1;
      if (O < 1)
        throw new Error();
      this.month = O % 12 || O;
    }], Y: [/[+-]?\d+/, c("year")], YY: [a, function(x) {
      this.year = h(x);
    }], YYYY: [/\d{4}/, c("year")], Z: g, ZZ: g };
    function I(x) {
      var O, E;
      O = x, E = s && s.formats;
      for (var k = (x = O.replace(/(\[[^\]]+])|(LTS?|l{1,4}|L{1,4})/g, function(R, N, M) {
        var z = M && M.toUpperCase();
        return N || E[M] || u[M] || E[z].replace(/(\[[^\]]+])|(MMMM|MM|DD|dddd)/g, function(J, j, y) {
          return j || y.slice(1);
        });
      })).match(e), w = k.length, V = 0; V < w; V += 1) {
        var A = k[V], P = T[A], l = P && P[0], r = P && P[1];
        k[V] = r ? { regex: l, parser: r } : A.replace(/^\[|\]$/g, "");
      }
      return function(R) {
        for (var N = {}, M = 0, z = 0; M < w; M += 1) {
          var J = k[M];
          if (typeof J == "string")
            z += J.length;
          else {
            var j = J.regex, y = J.parser, _ = R.slice(z), Y = j.exec(_)[0];
            y.call(N, Y), R = R.replace(Y, "");
          }
        }
        return function(S) {
          var B = S.afternoon;
          if (B !== void 0) {
            var F = S.hours;
            B ? F < 12 && (S.hours += 12) : F === 12 && (S.hours = 0), delete S.afternoon;
          }
        }(N), N;
      };
    }
    return function(x, O, E) {
      E.p.customParseFormat = !0, x && x.parseTwoDigitYear && (h = x.parseTwoDigitYear);
      var k = O.prototype, w = k.parse;
      k.parse = function(V) {
        var A = V.date, P = V.utc, l = V.args;
        this.$u = P;
        var r = l[1];
        if (typeof r == "string") {
          var R = l[2] === !0, N = l[3] === !0, M = R || N, z = l[2];
          N && (z = l[2]), s = this.$locale(), !R && z && (s = E.Ls[z]), this.$d = function(_, Y, S) {
            try {
              if (["x", "X"].indexOf(Y) > -1)
                return new Date((Y === "X" ? 1e3 : 1) * _);
              var B = I(Y)(_), F = B.year, G = B.month, Q = B.day, te = B.hours, re = B.minutes, de = B.seconds, we = B.milliseconds, oe = B.zone, be = /* @__PURE__ */ new Date(), he = Q || (F || G ? 1 : be.getDate()), ne = F || be.getFullYear(), fe = 0;
              F && !G || (fe = G > 0 ? G - 1 : be.getMonth());
              var ge = te || 0, ke = re || 0, Re = de || 0, Ie = we || 0;
              return oe ? new Date(Date.UTC(ne, fe, he, ge, ke, Re, Ie + 60 * oe.offset * 1e3)) : S ? new Date(Date.UTC(ne, fe, he, ge, ke, Re, Ie)) : new Date(ne, fe, he, ge, ke, Re, Ie);
            } catch {
              return /* @__PURE__ */ new Date("");
            }
          }(A, r, P), this.init(), z && z !== !0 && (this.$L = this.locale(z).$L), M && A != this.format(r) && (this.$d = /* @__PURE__ */ new Date("")), s = {};
        } else if (r instanceof Array)
          for (var J = r.length, j = 1; j <= J; j += 1) {
            l[1] = r[j - 1];
            var y = E.apply(this, l);
            if (y.isValid()) {
              this.$d = y.$d, this.$L = y.$L, this.init();
              break;
            }
            j === J && (this.$d = /* @__PURE__ */ new Date(""));
          }
        else
          w.call(this, V);
      };
    };
  });
})(Yt);
var jr = Yt.exports;
const Er = /* @__PURE__ */ je(jr);
var Ct = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    return function(u, e, a) {
      e.prototype.isToday = function() {
        var o = "YYYY-MM-DD", m = a();
        return this.format(o) === m.format(o);
      };
    };
  });
})(Ct);
var Lr = Ct.exports;
const Fr = /* @__PURE__ */ je(Lr);
var jt = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    return function(u, e, a) {
      e.prototype.isBetween = function(o, m, s, h) {
        var c = a(o), g = a(m), b = (h = h || "()")[0] === "(", D = h[1] === ")";
        return (b ? this.isAfter(c, s) : !this.isBefore(c, s)) && (D ? this.isBefore(g, s) : !this.isAfter(g, s)) || (b ? this.isBefore(c, s) : !this.isAfter(c, s)) && (D ? this.isAfter(g, s) : !this.isBefore(g, s));
      };
    };
  });
})(jt);
var Br = jt.exports;
const Nr = /* @__PURE__ */ je(Br);
var Et = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    var u, e, a = 1e3, o = 6e4, m = 36e5, s = 864e5, h = /\[([^\]]+)]|Y{1,4}|M{1,4}|D{1,2}|d{1,4}|H{1,2}|h{1,2}|a|A|m{1,2}|s{1,2}|Z{1,2}|SSS/g, c = 31536e6, g = 2628e6, b = /^(-|\+)?P(?:([-+]?[0-9,.]*)Y)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)W)?(?:([-+]?[0-9,.]*)D)?(?:T(?:([-+]?[0-9,.]*)H)?(?:([-+]?[0-9,.]*)M)?(?:([-+]?[0-9,.]*)S)?)?$/, D = { years: c, months: g, days: s, hours: m, minutes: o, seconds: a, milliseconds: 1, weeks: 6048e5 }, T = function(P) {
      return P instanceof V;
    }, I = function(P, l, r) {
      return new V(P, r, l.$l);
    }, x = function(P) {
      return e.p(P) + "s";
    }, O = function(P) {
      return P < 0;
    }, E = function(P) {
      return O(P) ? Math.ceil(P) : Math.floor(P);
    }, k = function(P) {
      return Math.abs(P);
    }, w = function(P, l) {
      return P ? O(P) ? { negative: !0, format: "" + k(P) + l } : { negative: !1, format: "" + P + l } : { negative: !1, format: "" };
    }, V = function() {
      function P(r, R, N) {
        var M = this;
        if (this.$d = {}, this.$l = N, r === void 0 && (this.$ms = 0, this.parseFromMilliseconds()), R)
          return I(r * D[x(R)], this);
        if (typeof r == "number")
          return this.$ms = r, this.parseFromMilliseconds(), this;
        if (typeof r == "object")
          return Object.keys(r).forEach(function(j) {
            M.$d[x(j)] = r[j];
          }), this.calMilliseconds(), this;
        if (typeof r == "string") {
          var z = r.match(b);
          if (z) {
            var J = z.slice(2).map(function(j) {
              return j != null ? Number(j) : 0;
            });
            return this.$d.years = J[0], this.$d.months = J[1], this.$d.weeks = J[2], this.$d.days = J[3], this.$d.hours = J[4], this.$d.minutes = J[5], this.$d.seconds = J[6], this.calMilliseconds(), this;
          }
        }
        return this;
      }
      var l = P.prototype;
      return l.calMilliseconds = function() {
        var r = this;
        this.$ms = Object.keys(this.$d).reduce(function(R, N) {
          return R + (r.$d[N] || 0) * D[N];
        }, 0);
      }, l.parseFromMilliseconds = function() {
        var r = this.$ms;
        this.$d.years = E(r / c), r %= c, this.$d.months = E(r / g), r %= g, this.$d.days = E(r / s), r %= s, this.$d.hours = E(r / m), r %= m, this.$d.minutes = E(r / o), r %= o, this.$d.seconds = E(r / a), r %= a, this.$d.milliseconds = r;
      }, l.toISOString = function() {
        var r = w(this.$d.years, "Y"), R = w(this.$d.months, "M"), N = +this.$d.days || 0;
        this.$d.weeks && (N += 7 * this.$d.weeks);
        var M = w(N, "D"), z = w(this.$d.hours, "H"), J = w(this.$d.minutes, "M"), j = this.$d.seconds || 0;
        this.$d.milliseconds && (j += this.$d.milliseconds / 1e3, j = Math.round(1e3 * j) / 1e3);
        var y = w(j, "S"), _ = r.negative || R.negative || M.negative || z.negative || J.negative || y.negative, Y = z.format || J.format || y.format ? "T" : "", S = (_ ? "-" : "") + "P" + r.format + R.format + M.format + Y + z.format + J.format + y.format;
        return S === "P" || S === "-P" ? "P0D" : S;
      }, l.toJSON = function() {
        return this.toISOString();
      }, l.format = function(r) {
        var R = r || "YYYY-MM-DDTHH:mm:ss", N = { Y: this.$d.years, YY: e.s(this.$d.years, 2, "0"), YYYY: e.s(this.$d.years, 4, "0"), M: this.$d.months, MM: e.s(this.$d.months, 2, "0"), D: this.$d.days, DD: e.s(this.$d.days, 2, "0"), H: this.$d.hours, HH: e.s(this.$d.hours, 2, "0"), m: this.$d.minutes, mm: e.s(this.$d.minutes, 2, "0"), s: this.$d.seconds, ss: e.s(this.$d.seconds, 2, "0"), SSS: e.s(this.$d.milliseconds, 3, "0") };
        return R.replace(h, function(M, z) {
          return z || String(N[M]);
        });
      }, l.as = function(r) {
        return this.$ms / D[x(r)];
      }, l.get = function(r) {
        var R = this.$ms, N = x(r);
        return N === "milliseconds" ? R %= 1e3 : R = N === "weeks" ? E(R / D[N]) : this.$d[N], R || 0;
      }, l.add = function(r, R, N) {
        var M;
        return M = R ? r * D[x(R)] : T(r) ? r.$ms : I(r, this).$ms, I(this.$ms + M * (N ? -1 : 1), this);
      }, l.subtract = function(r, R) {
        return this.add(r, R, !0);
      }, l.locale = function(r) {
        var R = this.clone();
        return R.$l = r, R;
      }, l.clone = function() {
        return I(this.$ms, this);
      }, l.humanize = function(r) {
        return u().add(this.$ms, "ms").locale(this.$l).fromNow(!r);
      }, l.valueOf = function() {
        return this.asMilliseconds();
      }, l.milliseconds = function() {
        return this.get("milliseconds");
      }, l.asMilliseconds = function() {
        return this.as("milliseconds");
      }, l.seconds = function() {
        return this.get("seconds");
      }, l.asSeconds = function() {
        return this.as("seconds");
      }, l.minutes = function() {
        return this.get("minutes");
      }, l.asMinutes = function() {
        return this.as("minutes");
      }, l.hours = function() {
        return this.get("hours");
      }, l.asHours = function() {
        return this.as("hours");
      }, l.days = function() {
        return this.get("days");
      }, l.asDays = function() {
        return this.as("days");
      }, l.weeks = function() {
        return this.get("weeks");
      }, l.asWeeks = function() {
        return this.as("weeks");
      }, l.months = function() {
        return this.get("months");
      }, l.asMonths = function() {
        return this.as("months");
      }, l.years = function() {
        return this.get("years");
      }, l.asYears = function() {
        return this.as("years");
      }, P;
    }(), A = function(P, l, r) {
      return P.add(l.years() * r, "y").add(l.months() * r, "M").add(l.days() * r, "d").add(l.hours() * r, "h").add(l.minutes() * r, "m").add(l.seconds() * r, "s").add(l.milliseconds() * r, "ms");
    };
    return function(P, l, r) {
      u = r, e = r().$utils(), r.duration = function(M, z) {
        var J = r.locale();
        return I(M, { $l: J }, z);
      }, r.isDuration = T;
      var R = l.prototype.add, N = l.prototype.subtract;
      l.prototype.add = function(M, z) {
        return T(M) ? A(this, M, 1) : R.bind(this)(M, z);
      }, l.prototype.subtract = function(M, z) {
        return T(M) ? A(this, M, -1) : N.bind(this)(M, z);
      };
    };
  });
})(Et);
var Rr = Et.exports;
const Ir = /* @__PURE__ */ je(Rr);
var Lt = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    var u = "week", e = "year";
    return function(a, o, m) {
      var s = o.prototype;
      s.week = function(h) {
        if (h === void 0 && (h = null), h !== null)
          return this.add(7 * (h - this.week()), "day");
        var c = this.$locale().yearStart || 1;
        if (this.month() === 11 && this.date() > 25) {
          var g = m(this).startOf(e).add(1, e).date(c), b = m(this).endOf(u);
          if (g.isBefore(b))
            return 1;
        }
        var D = m(this).startOf(e).date(c).startOf(u).subtract(1, "millisecond"), T = this.diff(D, u, !0);
        return T < 0 ? m(this).startOf("week").week() : Math.ceil(T);
      }, s.weeks = function(h) {
        return h === void 0 && (h = null), this.week(h);
      };
    };
  });
})(Lt);
var Hr = Lt.exports;
const Ur = /* @__PURE__ */ je(Hr);
var Wr = { exports: {} };
(function(t, n) {
  (function(u, e) {
    t.exports = e();
  })(Te, function() {
    return { name: "en", weekdays: "Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday".split("_"), months: "January_February_March_April_May_June_July_August_September_October_November_December".split("_"), ordinal: function(u) {
      var e = ["th", "st", "nd", "rd"], a = u % 100;
      return "[" + u + (e[(a - 20) % 10] || e[a] || e[0]) + "]";
    } };
  });
})(Wr);
const Kr = { class: "flex justify-between items-center px-2 py-1.5" }, zr = { class: "flex-shrink-0" }, Gr = { class: "inline-flex rounded-full" }, Zr = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Jr = ["d"], qr = { class: "px-1.5 space-x-1.5 flex flex-1" }, Xr = { class: "flex-1 flex rounded-md" }, Qr = ["textContent"], en = { class: "flex-1 flex rounded-md" }, tn = ["textContent"], rn = { class: "flex-shrink-0" }, nn = { class: "inline-flex rounded-full" }, an = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, on = ["d"], Ze = /* @__PURE__ */ ie({
  __name: "Header",
  props: {
    panel: {},
    calendar: {}
  },
  setup(t) {
    return (n, u) => (W(), Z("div", Kr, [
      C("div", zr, [
        ve(C("span", Gr, [
          C("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: u[0] || (u[0] = (e) => n.panel.calendar ? n.calendar.onPrevious() : n.calendar.onPreviousYear())
          }, [
            (W(), Z("svg", Zr, [
              C("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: n.panel.calendar ? "M15 19l-7-7 7-7" : "M11 19l-7-7 7-7m8 14l-7-7 7-7"
              }, null, 8, Jr)
            ]))
          ])
        ], 512), [
          [pe, n.panel.calendar || n.panel.year]
        ])
      ]),
      C("div", qr, [
        C("span", Xr, [
          C("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: u[1] || (u[1] = (e) => n.calendar.openMonth()),
            textContent: ae(n.calendar.month)
          }, null, 8, Qr)
        ]),
        C("span", en, [
          C("button", {
            type: "button",
            class: "px-3 py-1.5 block w-full leading-relaxed rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-semibold sm:font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: u[2] || (u[2] = (e) => n.calendar.openYear()),
            textContent: ae(n.calendar.year)
          }, null, 8, tn)
        ])
      ]),
      C("div", rn, [
        ve(C("span", nn, [
          C("button", {
            type: "button",
            class: "p-1.5 rounded-full bg-white text-vtd-secondary-600 transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50",
            onClick: u[3] || (u[3] = (e) => n.panel.calendar ? n.calendar.onNext() : n.calendar.onNextYear())
          }, [
            (W(), Z("svg", an, [
              C("path", {
                "stroke-linecap": "round",
                "stroke-linejoin": "round",
                "stroke-width": "1.5",
                d: n.panel.calendar ? "M9 5l7 7-7 7" : "M13 5l7 7-7 7M5 5l7 7-7 7"
              }, null, 8, on)
            ]))
          ])
        ], 512), [
          [pe, n.panel.calendar || n.panel.year]
        ])
      ])
    ]));
  }
});
function $e(t, n) {
  const u = Oe(t, n);
  if (!u)
    throw new Error(`Could not resolve ${t.description}`);
  return u;
}
const Ft = Symbol("isBetweenRange"), Bt = Symbol(
  "betweenRangeClasses"
), Nt = Symbol("datepickerClasses"), Rt = Symbol("atMouseOver"), It = Symbol("setToToday"), Ht = Symbol("setToYesterday"), Ut = Symbol("setToLastDay"), Wt = Symbol("setToThisMonth"), Kt = Symbol("setToLastMonth"), zt = Symbol("setToCustomShortcut"), sn = {
  key: 0,
  class: "relative w-full border-t border-b-0 sm:border-t-0 sm:border-b lg:border-b-0 lg:border-r border-black/[.1] order-last sm:order-none dark:border-vtd-secondary-700/[1] sm:mt-1 lg:mr-1 sm:mb-1 lg:mb-0 sm:mx-1 lg:mx-0 sm:w-auto"
}, ln = {
  key: 0,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, un = ["onClick", "textContent"], dn = {
  key: 1,
  class: "grid grid-cols-2 sm:grid-cols-3 gap-1 lg:block w-full pr-5 sm:pr-6 mt-1.5 sm:mt-0 sm:mb-1.5 lg:mb-0"
}, vt = /* @__PURE__ */ ie({
  __name: "Shortcut",
  props: {
    shortcuts: { type: [Boolean, Function] },
    close: { type: Function },
    asRange: { type: Boolean },
    asSingle: { type: Boolean },
    i18n: {}
  },
  setup(t) {
    const n = t, u = $e(It), e = $e(Ht), a = $e(Ut), o = $e(Wt), m = $e(Kt), s = $e(zt), h = ue(() => typeof n.shortcuts == "function" ? n.shortcuts() : !1);
    return (c, g) => n.asRange && n.asSingle || n.asRange && !n.asSingle ? (W(), Z("div", sn, [
      h.value ? (W(), Z("ol", ln, [
        (W(!0), Z(xe, null, ze(h.value, (b, D) => (W(), Z("li", { key: D }, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: Se((T) => le(s)(b, c.close), ["prevent"]),
            textContent: ae(b.label)
          }, null, 8, un)
        ]))), 128))
      ])) : (W(), Z("ol", dn, [
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[0] || (g[0] = Se((b) => le(u)(c.close), ["prevent"]))
          }, ae(n.i18n.today), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[1] || (g[1] = Se((b) => le(e)(c.close), ["prevent"]))
          }, ae(n.i18n.yesterday), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[2] || (g[2] = Se((b) => le(a)(7, c.close), ["prevent"]))
          }, ae(n.i18n.past(7)), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[3] || (g[3] = Se((b) => le(a)(30, c.close), ["prevent"]))
          }, ae(n.i18n.past(30)), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[4] || (g[4] = Se((b) => le(o)(c.close), ["prevent"]))
          }, ae(n.i18n.currentMonth), 1)
        ]),
        C("li", null, [
          C("a", {
            href: "#",
            class: "vtd-shortcuts block text-sm lg:text-xs px-2 py-2 sm:leading-4 whitespace-nowrap font-medium rounded text-vtd-primary-600 hover:text-vtd-primary-700 transition-colors hover:bg-vtd-secondary-100 focus:bg-vtd-secondary-100 focus:text-vtd-primary-600 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-primary-300 dark:text-vtd-primary-400 dark:focus:bg-vtd-secondary-700 dark:focus:text-vtd-primary-300",
            onClick: g[5] || (g[5] = Se((b) => le(m)(c.close), ["prevent"]))
          }, ae(n.i18n.pastMonth), 1)
        ])
      ]))
    ])) : ye("", !0);
  }
}), cn = { class: "grid grid-cols-7 gap-y-0.5 my-1" }, fn = {
  key: 0,
  class: "col-span-7 border-b relative dark:border-vtd-secondary-600"
}, vn = { class: "absolute -left-2 top-1/2 -translate-y-2/4 bg-white dark:bg-vtd-secondary-800 text-[8px] pr-2 text-vtd-secondary-400" }, pn = ["data-tooltip"], mn = ["disabled", "data-date", "onClick", "onMouseenter", "onFocusin", "textContent"], Je = /* @__PURE__ */ ie({
  __name: "Calendar",
  props: {
    calendar: {},
    weeks: {},
    weekNumber: { type: Boolean },
    asRange: { type: Boolean }
  },
  emits: ["updateDate"],
  setup(t, { emit: n }) {
    const u = n, e = $e(Ft), a = $e(Bt), o = $e(Nt), m = $e(Rt);
    return (s, h) => (W(), Z("div", cn, [
      ee(Xt, {
        "enter-from-class": "opacity-0",
        "enter-to-class": "opacity-100",
        "enter-active-class": "transition-opacity ease-out duration-300",
        "leave-active-class": "transition-opacity ease-in duration-200",
        "leave-from-class": "opacity-100",
        "leave-to-class": "opacity-0"
      }, {
        default: Fe(() => [
          (W(!0), Z(xe, null, ze(s.calendar.date(), (c, g) => (W(), Z(xe, { key: g }, [
            g % 7 === 0 && s.weekNumber ? (W(), Z("div", fn, [
              C("span", vn, ae(c.week()), 1)
            ])) : ye("", !0),
            C("div", {
              class: Ye(["relative", { "vtd-tooltip": s.asRange && c.duration() }]),
              "data-tooltip": `${c.duration()}`
            }, [
              ee(mt, {
                "enter-from-class": "opacity-0",
                "enter-to-class": "opacity-100",
                "enter-active-class": "transition-opacity ease-out duration-200",
                "leave-active-class": "transition-opacity ease-in duration-150",
                "leave-from-class": "opacity-100",
                "leave-to-class": "opacity-0"
              }, {
                default: Fe(() => [
                  le(e)(c) || c.hovered() ? (W(), Z("span", {
                    key: 0,
                    class: Ye(["absolute bg-vtd-primary-100 bg-opacity-60 dark:bg-vtd-secondary-700 dark:bg-opacity-50", le(a)(c)])
                  }, null, 2)) : ye("", !0)
                ]),
                _: 2
              }, 1024),
              C("button", {
                type: "button",
                class: Ye(["vtd-datepicker-date relative w-[2.7rem] h-[2.7rem] lg:w-10 lg:h-10 flex justify-center items-center text-xs 2xl:text-sm", [
                  le(o)(c),
                  s.asRange ? "transition-all" : "transition-colors"
                ]]),
                disabled: c.disabled || c.inRange(),
                "data-date": c.toDate(),
                onClick: (b) => u("updateDate", c),
                onMouseenter: (b) => le(m)(c),
                onFocusin: (b) => le(m)(c),
                textContent: ae(c.date())
              }, null, 42, mn)
            ], 10, pn)
          ], 64))), 128))
        ]),
        _: 1
      })
    ]));
  }
}), hn = { class: "flex flex-wrap" }, yn = { class: "flex rounded-md mt-1.5" }, bn = ["onClick", "textContent"], qe = /* @__PURE__ */ ie({
  __name: "Year",
  props: {
    years: {}
  },
  emits: ["updateYear"],
  setup(t, { emit: n }) {
    const u = n;
    return (e, a) => (W(), Z("div", hn, [
      (W(!0), Z(xe, null, ze(e.years, (o, m) => (W(), Z("div", {
        key: m,
        class: "w-1/2 px-0.5"
      }, [
        C("span", yn, [
          C("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            onClick: (s) => u("updateYear", o),
            textContent: ae(o)
          }, null, 8, bn)
        ])
      ]))), 128))
    ]));
  }
}), gn = { class: "grid grid-cols-7 py-2 mt-0.5" }, xn = ["textContent"], Xe = /* @__PURE__ */ ie({
  __name: "Week",
  props: {
    weeks: {}
  },
  setup(t) {
    return (n, u) => (W(), Z("div", gn, [
      (W(!0), Z(xe, null, ze(n.weeks, (e, a) => (W(), Z("div", {
        key: a,
        class: "text-vtd-secondary-500 text-xs 2xl:text-sm tracking-wide font-medium text-center cursor-default dark:text-vtd-secondary-400"
      }, [
        C("span", {
          textContent: ae(e)
        }, null, 8, xn)
      ]))), 128))
    ]));
  }
}), wn = { class: "flex flex-wrap mt-1.5" }, kn = { class: "flex rounded-md mt-1.5" }, $n = ["onClick", "textContent"], Qe = /* @__PURE__ */ ie({
  __name: "Month",
  props: {
    months: {}
  },
  emits: ["updateMonth"],
  setup(t, { emit: n }) {
    const u = n;
    return (e, a) => (W(), Z("div", wn, [
      (W(!0), Z(xe, null, ze(e.months, (o, m) => (W(), Z("div", {
        key: m,
        class: "w-1/2 px-0.5"
      }, [
        C("span", kn, [
          C("button", {
            type: "button",
            class: "px-3 py-2 block w-full leading-6 rounded-md bg-white text-xs 2xl:text-sm tracking-wide text-vtd-secondary-600 font-medium transition-colors border border-transparent hover:bg-vtd-secondary-100 hover:text-vtd-secondary-900 focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none uppercase dark:bg-vtd-secondary-800 dark:hover:bg-vtd-secondary-700 dark:text-vtd-secondary-300 dark:hover:text-vtd-secondary-100 dark:focus:bg-vtd-secondary-700",
            onClick: (s) => u("updateMonth", m),
            textContent: ae(o)
          }, null, 8, $n)
        ])
      ]))), 128))
    ]));
  }
});
function Mn() {
  const t = (s) => {
    const h = [], c = s.localeData().firstDayOfWeek();
    for (let g = 0; g <= s.date(0 - c).day(); g++)
      h.push(s.date(0).subtract(g, "day"));
    return h.sort((g, b) => g.date() - b.date());
  };
  return {
    usePreviousDate: t,
    useCurrentDate: (s) => Array.from(
      {
        length: s.daysInMonth()
      },
      (h, c) => s.date(c + 1)
    ),
    useNextDate: (s) => {
      const h = [];
      for (let c = 1; c <= 42 - (t(s).length + s.daysInMonth()); c++)
        h.push(s.date(c).month(s.month()).add(1, "month"));
      return h;
    },
    useDisableDate: (s, { disableDate: h }) => typeof h == "function" ? h(s.toDate()) : !1,
    useBetweenRange: (s, { previous: h, next: c }) => {
      const g = h.isAfter(c, "date") ? "(]" : "[)";
      return !!(s.isBetween(h, c, "date", g) && !s.off);
    },
    useToValueFromString: (s, { formatter: h }) => s.format(h.date),
    useToValueFromArray: ({ previous: s, next: h }, {
      formatter: c,
      separator: g
    }) => `${s.format(c.date)}${g}${h.format(
      c.date
    )}`
  };
}
function Sn() {
  return {
    useVisibleViewport: (n) => {
      if (n) {
        const { right: u } = n.getBoundingClientRect(), e = window.innerWidth || document.documentElement.clientWidth;
        return u > e;
      } else
        return null;
    }
  };
}
const _n = ["disabled", "placeholder", "onKeyup"], Dn = { class: "absolute inset-y-0 right-0 inline-flex items-center rounded-md overflow-hidden" }, On = ["disabled"], Pn = {
  class: "w-5 h-5",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, Vn = {
  key: 0,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M6 18L18 6M6 6l12 12"
}, Tn = {
  key: 1,
  "stroke-linecap": "round",
  "stroke-linejoin": "round",
  "stroke-width": "1.5",
  d: "M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
}, An = { class: "flex flex-wrap lg:flex-nowrap" }, Yn = { class: "relative flex flex-wrap sm:flex-nowrap p-1 w-full" }, Cn = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, jn = /* @__PURE__ */ C("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), En = [
  jn
], Ln = { class: "px-0.5 sm:px-2" }, Fn = {
  key: 1,
  class: "relative w-full md:w-1/2 lg:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, Bn = { class: "px-0.5 sm:px-2" }, Nn = { key: 0 }, Rn = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, In = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, Hn = ["disabled", "onClick", "textContent"], Un = ["onClick", "textContent"], Wn = {
  key: 1,
  class: "sm:hidden"
}, Kn = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, zn = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, Gn = ["onClick", "textContent"], Zn = {
  key: 1,
  class: "flex"
}, Jn = { class: "bg-white rounded-lg shadow-sm border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]" }, qn = { class: "flex flex-wrap lg:flex-nowrap" }, Xn = { class: "relative flex flex-wrap sm:flex-nowrap p-1 w-full" }, Qn = {
  key: 0,
  class: "hidden h-full absolute inset-0 sm:flex justify-center items-center"
}, ea = /* @__PURE__ */ C("div", { class: "h-full border-r border-black/[.1] dark:border-vtd-secondary-700/[1]" }, null, -1), ta = [
  ea
], ra = { class: "px-0.5 sm:px-2" }, na = {
  key: 1,
  class: "relative w-full md:w-1/2 lg:w-80 overflow-hidden mt-3 sm:mt-0 sm:ml-2"
}, aa = { class: "px-0.5 sm:px-2" }, oa = { key: 0 }, sa = { class: "mt-2 mx-2 py-1.5 border-t border-black/[.1] dark:border-vtd-secondary-700/[1]" }, la = { class: "mt-1.5 sm:flex sm:flex-row-reverse" }, ua = ["disabled", "textContent"], ia = /* @__PURE__ */ ie({
  __name: "VueTailwindDatePicker",
  props: {
    noInput: { type: Boolean },
    overlay: { type: Boolean },
    asSingle: { type: Boolean },
    useRange: { type: Boolean },
    placeholder: { default: "" },
    i18n: { default: "en" },
    inputClasses: { default: "" },
    disabled: { type: Boolean, default: !1 },
    disableInRange: { type: Boolean, default: !1 },
    disableDate: { type: [Boolean, Function], default: !1 },
    autoApply: { type: Boolean, default: !0 },
    shortcuts: { type: [Boolean, Function], default: !0 },
    separator: { default: " ~ " },
    formatter: { default: () => ({
      date: "YYYY-MM-DD HH:mm:ss",
      month: "MMM"
    }) },
    startFrom: { default: () => /* @__PURE__ */ new Date() },
    weekdaysSize: { default: "short" },
    weekNumber: { type: Boolean, default: !1 },
    options: { default: () => ({
      shortcuts: {
        today: "Today",
        yesterday: "Yesterday",
        past: (t) => `Last ${t} Days`,
        currentMonth: "This Month",
        pastMonth: "Last Month"
      },
      footer: {
        apply: "Apply",
        cancel: "Cancel"
      }
    }) },
    modelValue: { default: () => [/* @__PURE__ */ new Date(), /* @__PURE__ */ new Date()] }
  },
  emits: ["update:modelValue", "selectMonth", "selectYear", "selectRightMonth", "selectRightYear", "clickPrev", "clickNext", "clickRightPrev", "clickRightNext"],
  setup(t, { expose: n, emit: u }) {
    const e = t, a = u, {
      useCurrentDate: o,
      useDisableDate: m,
      useBetweenRange: s,
      useNextDate: h,
      usePreviousDate: c,
      useToValueFromArray: g,
      useToValueFromString: b
    } = Mn(), { useVisibleViewport: D } = Sn();
    d.extend(Ar), d.extend(Cr), d.extend(Er), d.extend(Fr), d.extend(Nr), d.extend(Ir), d.extend(Ur);
    const T = q(null), I = q(null), x = q(null), O = q(""), E = q(null), k = q(""), w = q([]), V = q([]), A = q(null), P = q(null), l = pt({
      previous: {
        calendar: !0,
        month: !1,
        year: !1
      },
      next: {
        calendar: !0,
        month: !1,
        year: !1
      }
    }), r = q({
      previous: d(),
      next: d().add(1, "month"),
      year: {
        previous: d().year(),
        next: d().year()
      },
      weeks: e.weekdaysSize === "min" ? d.weekdaysMin() : d.weekdaysShort(),
      months: e.formatter.month === "MMM" ? d.monthsShort() : d.months()
    }), R = ue(() => r.value.weeks), N = ue(() => r.value.months), M = ue(() => {
      const { previous: p, next: f, year: v } = le(r);
      return {
        previous: {
          date: () => c(p).concat(o(p)).concat(h(p)).map((i) => (Object.assign(i, {
            today: i.isToday(),
            active: p.month() === i.month(),
            off: p.month() !== i.month(),
            sunday: i.day() === 0,
            disabled: m(i, e) && !_(i),
            inRange: () => {
              if (e.asSingle && !e.useRange)
                return p.month() !== i.month();
            },
            hovered: () => y() && w.value.length > 1 ? (i.isBetween(
              w.value[0],
              w.value[1],
              "date",
              "()"
            ) || i.isBetween(
              w.value[1],
              w.value[0],
              "date",
              "(]"
            )) && p.month() === i.month() : !1,
            duration: () => !1
          }), i)),
          month: p && p.format(e.formatter.month),
          year: p && p.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (i, $) => v.previous + $
          ),
          onPrevious: () => {
            r.value.previous = p.subtract(1, "month"), a("clickPrev", r.value.previous);
          },
          onNext: () => {
            r.value.previous = p.add(1, "month"), p.diff(f, "month") === -1 && (r.value.next = f.add(1, "month")), a("clickNext", r.value.previous);
          },
          onPreviousYear: () => {
            r.value.year.previous = r.value.year.previous - 12;
          },
          onNextYear: () => {
            r.value.year.previous = r.value.year.previous + 12;
          },
          openMonth: () => {
            l.previous.month = !l.previous.month, l.previous.year = !1, l.previous.calendar = !l.previous.month;
          },
          setMonth: (i) => {
            r.value.previous = p.month(i), l.previous.month = !l.previous.month, l.previous.year = !1, l.previous.calendar = !l.previous.month, a("selectMonth", r.value.previous), He(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.next = r.value.next.year();
            });
          },
          openYear: () => {
            l.previous.year = !l.previous.year, l.previous.month = !1, l.previous.calendar = !l.previous.year;
          },
          setYear: (i) => {
            r.value.previous = p.year(i), l.previous.year = !l.previous.year, l.previous.calendar = !l.previous.year, a("selectYear", r.value.previous), He(() => {
              (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month")), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            });
          }
        },
        next: {
          date: () => c(f).concat(o(f)).concat(h(f)).map((i) => (Object.assign(i, {
            today: i.isToday(),
            active: f.month() === i.month(),
            off: f.month() !== i.month(),
            sunday: i.day() === 0,
            disabled: m(i, e) && !_(i),
            inRange: () => {
              if (e.asSingle && !e.useRange)
                return f.month() !== i.month();
            },
            hovered: () => w.value.length > 1 ? (i.isBetween(
              w.value[0],
              w.value[1],
              "date",
              "()"
            ) || i.isBetween(
              w.value[1],
              w.value[0],
              "date",
              "(]"
            )) && f.month() === i.month() : !1,
            duration: () => !1
          }), i)),
          month: f && f.format(e.formatter.month),
          year: f && f.year(),
          years: () => Array.from(
            {
              length: 12
            },
            (i, $) => v.next + $
          ),
          onPrevious: () => {
            r.value.next = f.subtract(1, "month"), f.diff(p, "month") === 1 && (r.value.previous = p.subtract(1, "month")), a("clickRightPrev", r.value.next);
          },
          onNext: () => {
            r.value.next = f.add(1, "month"), a("clickRightNext", r.value.next);
          },
          onPreviousYear: () => {
            r.value.year.next = r.value.year.next - 12;
          },
          onNextYear: () => {
            r.value.year.next = r.value.year.next + 12;
          },
          openMonth: () => {
            l.next.month = !l.next.month, l.next.year = !1, l.next.calendar = !l.next.month;
          },
          setMonth: (i) => {
            r.value.next = f.month(i), l.next.month = !l.next.month, l.next.year = !1, l.next.calendar = !l.next.month, a("selectRightMonth", r.value.next), He(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(
                1,
                "month"
              )), r.value.year.previous = r.value.previous.year();
            });
          },
          openYear: () => {
            l.next.year = !l.next.year, l.next.month = !1, l.next.calendar = !l.next.year;
          },
          setYear: (i) => {
            r.value.next = f.year(i), l.next.year = !l.next.year, l.next.month = !1, l.next.calendar = !l.next.year, a("selectRightYear", r.value.next), He(() => {
              (r.value.previous.isSame(r.value.next, "month") || r.value.previous.isAfter(r.value.next)) && (r.value.previous = r.value.next.subtract(
                1,
                "month"
              )), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year();
            });
          }
        }
      };
    }), z = q(!1);
    setTimeout(() => {
      z.value = !0;
    }, 250);
    function J() {
      return d().localeData().firstDayOfWeek();
    }
    function j(p) {
      const f = [...p], v = f.shift();
      return [...f, v];
    }
    function y() {
      return !e.useRange && !e.asSingle ? !0 : !e.useRange && e.asSingle ? !1 : e.useRange && !e.asSingle ? !0 : !!(e.useRange && e.asSingle);
    }
    function _(p) {
      if (e.disableInRange || k.value === "")
        return !1;
      let f, v;
      if (Array.isArray(e.modelValue)) {
        const [i, $] = e.modelValue;
        f = i, v = $;
      } else if (typeof e.modelValue == "object") {
        if (e.modelValue) {
          const [i, $] = Object.values(e.modelValue);
          f = i, v = $;
        }
      } else {
        const [i, $] = e.modelValue.split(e.separator);
        f = i, v = $;
      }
      return p.isBetween(
        d(f, e.formatter.date, !0),
        d(v, e.formatter.date, !0),
        "date",
        "[]"
      );
    }
    function Y() {
      A.value = null, P.value = null, w.value = [], E.value = null;
    }
    function S() {
      if (k.value = "", Array.isArray(e.modelValue))
        a("update:modelValue", []);
      else if (typeof e.modelValue == "object") {
        const p = {}, [f, v] = Object.keys(e.modelValue);
        p[f] = "", p[v] = "", a("update:modelValue", p);
      } else
        a("update:modelValue", "");
      V.value = [], I.value && I.value.focus();
    }
    n({ clearPicker: S });
    function B() {
      if (y()) {
        const [p, f] = k.value.split(e.separator), [v, i] = [
          d(p, e.formatter.date, !0),
          d(f, e.formatter.date, !0)
        ];
        if (v.isValid() && i.isValid())
          if (F(v), F(i), Array.isArray(e.modelValue))
            a("update:modelValue", [p, f]);
          else if (typeof e.modelValue == "object") {
            const $ = {}, [L, K] = Object.keys(e.modelValue);
            $[L] = p, $[K] = f, a("update:modelValue", $);
          } else
            a(
              "update:modelValue",
              g(
                {
                  previous: v,
                  next: i
                },
                e
              )
            );
      } else {
        const p = d(k.value, e.formatter.date, !0);
        if (p.isValid())
          if (F(p), Array.isArray(e.modelValue))
            a("update:modelValue", [k.value]);
          else if (typeof e.modelValue == "object") {
            const f = {}, [v] = Object.keys(e.modelValue);
            f[v] = k.value, a("update:modelValue", f);
          } else
            a("update:modelValue", k.value);
      }
    }
    function F(p, f) {
      if (y())
        if (A.value)
          if (P.value = p, e.autoApply) {
            p.isBefore(A.value) ? k.value = g(
              {
                previous: p,
                next: A.value
              },
              e
            ) : k.value = g(
              {
                previous: A.value,
                next: p
              },
              e
            );
            const [v, i] = k.value.split(e.separator);
            if (Array.isArray(e.modelValue))
              a("update:modelValue", [
                d(v, e.formatter.date, !0).format(e.formatter.date),
                d(i, e.formatter.date, !0).format(e.formatter.date)
              ]);
            else if (typeof e.modelValue == "object") {
              const $ = {}, [L, K] = Object.keys(e.modelValue);
              $[L] = v, $[K] = i, a("update:modelValue", $);
            } else
              a(
                "update:modelValue",
                g(
                  {
                    previous: d(v, e.formatter.date, !0),
                    next: d(i, e.formatter.date, !0)
                  },
                  e
                )
              );
            f && f(), V.value = [], d(v, e.formatter.date, !0).isSame(
              d(i, e.formatter.date, !0),
              "month"
            ) || (r.value.previous = d(v, e.formatter.date, !0), r.value.next = d(i, e.formatter.date, !0)), Y();
          } else {
            A.value.isAfter(p, "month") ? V.value = [p, A.value] : V.value = [A.value, p];
            const [v, i] = V.value;
            v.isSame(i, "month") || (r.value.previous = v, r.value.next = i), Y();
          }
        else
          V.value = [], A.value = p, E.value = p, w.value.push(p), V.value.push(p), r.value.previous = p, r.value.next.isSame(p, "month") && (r.value.previous = r.value.next, r.value.next = p.add(1, "month"));
      else if (e.autoApply) {
        if (k.value = b(p, e), Array.isArray(e.modelValue))
          a("update:modelValue", [k.value]);
        else if (typeof e.modelValue == "object") {
          const v = {}, [i] = Object.keys(e.modelValue);
          v[i] = k.value, a("update:modelValue", v);
        } else
          a("update:modelValue", k.value);
        f && f(), V.value = [], Y();
      } else
        V.value = [p], Y();
    }
    function G(p) {
      if (V.value.length < 1)
        return !1;
      let f;
      if (y()) {
        const [v, i] = V.value;
        i.isBefore(v) ? f = g(
          {
            previous: i,
            next: v
          },
          e
        ) : f = g(
          {
            previous: v,
            next: i
          },
          e
        );
      } else {
        const [v] = V.value;
        f = v;
      }
      if (y()) {
        const [v, i] = f.split(e.separator);
        if (Array.isArray(e.modelValue))
          a("update:modelValue", [
            d(v, e.formatter.date, !0).format(e.formatter.date),
            d(i, e.formatter.date, !0).format(e.formatter.date)
          ]);
        else if (typeof e.modelValue == "object") {
          const $ = {}, [L, K] = Object.keys(e.modelValue);
          $[L] = v, $[K] = i, a("update:modelValue", $);
        } else
          a(
            "update:modelValue",
            g(
              {
                previous: d(v, e.formatter.date, !0),
                next: d(i, e.formatter.date, !0)
              },
              e
            )
          );
        k.value = f;
      } else if (k.value = f.format(e.formatter.date), Array.isArray(e.modelValue))
        a("update:modelValue", [k.value]);
      else if (typeof e.modelValue == "object") {
        const v = {}, [i] = Object.keys(e.modelValue);
        v[i] = k.value, a("update:modelValue", v);
      } else
        a("update:modelValue", k.value);
      p && p();
    }
    function Q(p) {
      if (!y())
        return !1;
      if (A.value)
        w.value = [A.value, p];
      else
        return w.value = [], !1;
    }
    function te(p) {
      if (A.value && e.autoApply)
        return !1;
      let f, v;
      if (w.value.length > 1) {
        const [i, $] = w.value;
        f = d(i, e.formatter.date, !0), v = d($, e.formatter.date, !0);
      } else if (Array.isArray(e.modelValue))
        if (e.autoApply) {
          const [i, $] = e.modelValue;
          f = i && d(i, e.formatter.date, !0), v = $ && d($, e.formatter.date, !0);
        } else {
          const [i, $] = V.value;
          f = d(i, e.formatter.date, !0), v = d($, e.formatter.date, !0);
        }
      else if (typeof e.modelValue == "object")
        if (e.autoApply) {
          if (e.modelValue) {
            const [i, $] = Object.values(e.modelValue);
            f = i && d(i, e.formatter.date, !0), v = $ && d($, e.formatter.date, !0);
          }
        } else {
          const [i, $] = V.value;
          f = d(i, e.formatter.date, !0), v = d($, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        const [i, $] = e.modelValue ? e.modelValue.split(e.separator) : [null, null];
        f = i && d(i, e.formatter.date, !0), v = $ && d($, e.formatter.date, !0);
      } else {
        const [i, $] = V.value;
        f = d(i, e.formatter.date, !0), v = d($, e.formatter.date, !0);
      }
      return f && v ? s(p, {
        previous: f,
        next: v
      }) : !1;
    }
    function re(p) {
      const { today: f, active: v, off: i, disabled: $ } = p;
      let L, K, ce;
      if (y())
        if (Array.isArray(e.modelValue))
          if (E.value) {
            const [U, X] = w.value;
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [U, X] = e.modelValue;
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          } else {
            const [U, X] = V.value;
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          }
        else if (typeof e.modelValue == "object")
          if (E.value) {
            const [U, X] = w.value;
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          } else if (e.autoApply) {
            const [U, X] = e.modelValue ? Object.values(e.modelValue) : [null, null];
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          } else {
            const [U, X] = V.value;
            K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
          }
        else if (E.value) {
          const [U, X] = w.value;
          K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [U, X] = e.modelValue ? e.modelValue.split(e.separator) : [null, null];
          K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
        } else {
          const [U, X] = V.value;
          K = U && d(U, e.formatter.date, !0), ce = X && d(X, e.formatter.date, !0);
        }
      else if (Array.isArray(e.modelValue))
        if (e.autoApply) {
          if (e.modelValue.length > 0) {
            const [U] = e.modelValue;
            K = d(U, e.formatter.date, !0);
          }
        } else {
          const [U] = V.value;
          K = U && d(U, e.formatter.date, !0);
        }
      else if (typeof e.modelValue == "object")
        if (e.autoApply) {
          if (e.modelValue) {
            const [U] = Object.values(e.modelValue);
            K = d(U, e.formatter.date, !0);
          }
        } else {
          const [U] = V.value;
          K = U && d(U, e.formatter.date, !0);
        }
      else if (e.autoApply) {
        if (e.modelValue) {
          const [U] = e.modelValue.split(e.separator);
          K = d(U, e.formatter.date, !0);
        }
      } else {
        const [U] = V.value;
        K = U && d(U, e.formatter.date, !0);
      }
      return v && (L = f ? "text-vtd-primary-500 font-semibold dark:text-vtd-primary-400 rounded-full focus:bg-vtd-primary-50 focus:text-vtd-secondary-900 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:text-vtd-secondary-300 dark:hover:bg-vtd-secondary-700 dark:hover:text-vtd-secondary-300 dark:focus:bg-vtd-secondary-600 dark:focus:text-vtd-secondary-100 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-25 dark:focus:bg-opacity-50" : $ ? "text-vtd-secondary-600 font-normal disabled:text-vtd-secondary-500 disabled:cursor-not-allowed rounded-full" : p.isBetween(K, ce, "date", "()") ? "text-vtd-secondary-700 font-medium dark:text-vtd-secondary-100 rounded-full" : "text-vtd-secondary-600 font-medium dark:text-vtd-secondary-200 rounded-full"), i && (L = "text-vtd-secondary-400 font-light disabled:cursor-not-allowed"), K && ce && !i ? (p.isSame(K, "date") && (L = ce.isAfter(K, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed", K.isSame(ce, "date") && (L = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed")), p.isSame(ce, "date") && (L = ce.isAfter(K, "date") ? "bg-vtd-primary-500 text-white font-bold rounded-r-full disabled:cursor-not-allowed" : "bg-vtd-primary-500 text-white font-bold rounded-l-full disabled:cursor-not-allowed", K.isSame(ce, "date") && (L = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"))) : K && p.isSame(K, "date") && !i && (L = "bg-vtd-primary-500 text-white font-bold rounded-full disabled:cursor-not-allowed"), L;
    }
    function de(p) {
      let f, v, i;
      if (f = "", !y())
        return f;
      if (Array.isArray(e.modelValue))
        if (w.value.length > 1) {
          const [$, L] = w.value;
          v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
        } else if (e.autoApply) {
          const [$, L] = e.modelValue;
          v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
        } else {
          const [$, L] = V.value;
          v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
        }
      else if (typeof e.modelValue == "object")
        if (w.value.length > 1) {
          const [$, L] = w.value;
          v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
        } else if (e.autoApply) {
          if (e.modelValue) {
            const [$, L] = Object.values(e.modelValue);
            v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
          }
        } else {
          const [$, L] = V.value;
          v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
        }
      else if (w.value.length > 1) {
        const [$, L] = w.value;
        v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
      } else if (e.autoApply) {
        const [$, L] = e.modelValue ? e.modelValue.split(e.separator) : [null, null];
        v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
      } else {
        const [$, L] = V.value;
        v = $ && d($, e.formatter.date, !0), i = L && d(L, e.formatter.date, !0);
      }
      return v && i && (p.isSame(v, "date") ? (i.isBefore(v) && (f += " rounded-r-full inset-0"), v.isBefore(i) && (f += " rounded-l-full inset-0")) : p.isSame(i, "date") ? (i.isBefore(v) && (f += " rounded-l-full inset-0"), v.isBefore(i) && (f += " rounded-r-full inset-0")) : f += " inset-0"), f;
    }
    function we(p, f) {
      r.value.previous = d(p, e.formatter.date, !0), r.value.next = d(f, e.formatter.date, !0), (d.duration(r.value.next.diff(r.value.previous)).get("months") === 2 || d.duration(r.value.next.diff(r.value.previous)).get("months") === 1 && d.duration(r.value.next.diff(r.value.previous)).get("days") === 7) && (r.value.next = r.value.next.subtract(1, "month")), (r.value.next.isSame(r.value.previous, "month") || r.value.next.isBefore(r.value.previous)) && (r.value.next = r.value.previous.add(1, "month"));
    }
    function oe(p, f) {
      if (y())
        if (e.autoApply) {
          if (Array.isArray(e.modelValue))
            a("update:modelValue", [p, f]);
          else if (typeof e.modelValue == "object") {
            const v = {}, [i, $] = Object.keys(e.modelValue);
            v[i] = p, v[$] = f, a("update:modelValue", v);
          } else
            a(
              "update:modelValue",
              g(
                {
                  previous: d(p, e.formatter.date, !0),
                  next: d(f, e.formatter.date, !0)
                },
                e
              )
            );
          k.value = `${p}${e.separator}${f}`;
        } else
          V.value = [
            d(p, e.formatter.date, !0),
            d(f, e.formatter.date, !0)
          ];
      else if (e.autoApply) {
        if (Array.isArray(e.modelValue))
          a("update:modelValue", [p]);
        else if (typeof e.modelValue == "object") {
          const v = {}, [i] = Object.keys(e.modelValue);
          v[i] = p, a("update:modelValue", v);
        } else
          a("update:modelValue", p);
        k.value = p;
      } else
        V.value = [
          d(p, e.formatter.date, !0),
          d(f, e.formatter.date, !0)
        ];
      we(p, f);
    }
    function be(p) {
      const f = d().format(e.formatter.date), v = d().format(e.formatter.date);
      oe(f, v), p && p();
    }
    function he(p) {
      const f = d().subtract(1, "day").format(e.formatter.date), v = d().subtract(1, "day").format(e.formatter.date);
      oe(f, v), p && p();
    }
    function ne(p, f) {
      const v = d().subtract(p - 1, "day").format(e.formatter.date), i = d().format(e.formatter.date);
      oe(v, i), f && f();
    }
    function fe(p) {
      const f = d().date(1).format(e.formatter.date), v = d().date(d().daysInMonth()).format(e.formatter.date);
      oe(f, v), p && p();
    }
    function ge(p) {
      const f = d().date(1).subtract(1, "month").format(e.formatter.date), v = d().date(0).format(e.formatter.date);
      oe(f, v), p && p();
    }
    function ke(p, f) {
      const [v, i] = p.atClick(), $ = d(v).format(e.formatter.date), L = d(i).format(e.formatter.date);
      oe($, L), f && f();
    }
    Qt(
      () => V.value,
      (p) => {
        p.length > 0 && (l.previous.calendar = !0, l.previous.month = !1, l.previous.year = !1, l.next.calendar = !0, l.next.month = !1, l.next.year = !1);
      }
    ), Me(() => {
      e.placeholder ? O.value = e.placeholder : y() ? O.value = `${e.formatter.date}${e.separator}${e.formatter.date}` : O.value = e.formatter.date;
    }), Me(() => {
      const p = e.i18n, f = e.modelValue;
      He(async () => {
        d.locale(p);
        let v, i;
        if (y()) {
          if (Array.isArray(f)) {
            if (f.length > 0) {
              const [L, K] = f;
              v = d(L, e.formatter.date, !0), i = d(K, e.formatter.date, !0);
            }
          } else if (typeof f == "object") {
            if (!er(f))
              try {
                Object.keys(f);
              } catch {
                console.warn(
                  "[Vue Tailwind Datepicker]: It looks like you want to use Object as the argument %cv-model",
                  "font-style: italic; color: #42b883;",
                  ", but you pass it undefined or null."
                ), console.warn(
                  "[Vue Tailwind Datepicker]: We has replace with %c{ startDate: '', endDate: '' }",
                  "font-style: italic; color: #42b883;",
                  ", but you can replace manually."
                ), a("update:modelValue", {
                  startDate: "",
                  endDate: ""
                });
              }
            if (f) {
              const [L, K] = Object.values(f);
              v = L && d(L, e.formatter.date, !0), i = K && d(K, e.formatter.date, !0);
            }
          } else if (f) {
            const [L, K] = f.split(e.separator);
            v = d(L, e.formatter.date, !0), i = d(K, e.formatter.date, !0);
          }
          v && i ? (k.value = g(
            {
              previous: v,
              next: i
            },
            e
          ), i.isBefore(v, "month") ? (r.value.previous = i, r.value.next = v, r.value.year.previous = i.year(), r.value.year.next = v.year()) : i.isSame(v, "month") ? (r.value.previous = v, r.value.next = i.add(1, "month"), r.value.year.previous = v.year(), r.value.year.next = v.add(1, "year").year()) : (r.value.previous = v, r.value.next = i, r.value.year.previous = v.year(), r.value.year.next = i.year()), e.autoApply || (V.value = [v, i])) : (r.value.previous = d(e.startFrom), r.value.next = d(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
        } else {
          if (Array.isArray(f)) {
            if (f.length > 0) {
              const [L] = f;
              v = d(L, e.formatter.date, !0);
            }
          } else if (typeof f == "object") {
            if (f) {
              const [L] = Object.values(f);
              v = d(L, e.formatter.date, !0);
            }
          } else if (f.length) {
            const [L] = f.split(e.separator);
            v = d(L, e.formatter.date, !0);
          }
          v && v.isValid() ? (k.value = b(v, e), r.value.previous = v, r.value.next = v.add(1, "month"), r.value.year.previous = v.year(), r.value.year.next = v.add(1, "year").year(), e.autoApply || (V.value = [v])) : (r.value.previous = d(e.startFrom), r.value.next = d(e.startFrom).add(1, "month"), r.value.year.previous = r.value.previous.year(), r.value.year.next = r.value.next.year());
        }
        const $ = e.weekdaysSize === "min" ? d.weekdaysMin() : d.weekdaysShort();
        r.value.weeks = J() ? j($) : $, r.value.months = e.formatter.month === "MMM" ? d.monthsShort() : d.months();
      });
    });
    function Re(p) {
      return p && x.value === null && (x.value = D(T.value)), p && x.value ? "place-right" : "place-left";
    }
    function Ie(p) {
      return p && x.value === null && (x.value = D(T.value)), x.value ? "left-auto right-0" : "left-0 right-auto";
    }
    return se(Ft, te), se(Bt, de), se(Nt, re), se(Rt, Q), se(It, be), se(Ht, he), se(Ut, ne), se(Wt, fe), se(Kt, ge), se(zt, ke), (p, f) => e.noInput ? z.value ? (W(), Z("div", Zn, [
      C("div", Jn, [
        C("div", qn, [
          e.shortcuts ? (W(), Ue(vt, {
            key: 0,
            shortcuts: e.shortcuts,
            "as-range": y(),
            "as-single": e.asSingle,
            i18n: e.options.shortcuts
          }, null, 8, ["shortcuts", "as-range", "as-single", "i18n"])) : ye("", !0),
          C("div", Xn, [
            y() && !e.asSingle ? (W(), Z("div", Qn, ta)) : ye("", !0),
            C("div", {
              class: Ye(["relative w-full lg:w-80", {
                "mb-3 sm:mb-0 sm:mr-2 md:w-1/2": y() && !e.asSingle
              }])
            }, [
              ee(Ze, {
                panel: l.previous,
                calendar: M.value.previous
              }, null, 8, ["panel", "calendar"]),
              C("div", ra, [
                ve(ee(Qe, {
                  months: N.value,
                  onUpdateMonth: M.value.previous.setMonth
                }, null, 8, ["months", "onUpdateMonth"]), [
                  [pe, l.previous.month]
                ]),
                ve(ee(qe, {
                  years: M.value.previous.years(),
                  onUpdateYear: M.value.previous.setYear
                }, null, 8, ["years", "onUpdateYear"]), [
                  [pe, l.previous.year]
                ]),
                ve(C("div", null, [
                  ee(Xe, { weeks: R.value }, null, 8, ["weeks"]),
                  ee(Je, {
                    calendar: M.value.previous,
                    weeks: R.value,
                    "as-range": y(),
                    "week-number": p.weekNumber,
                    onUpdateDate: f[3] || (f[3] = (v) => F(v))
                  }, null, 8, ["calendar", "weeks", "as-range", "week-number"])
                ], 512), [
                  [pe, l.previous.calendar]
                ])
              ])
            ], 2),
            y() && !e.asSingle ? (W(), Z("div", na, [
              ee(Ze, {
                "as-prev-or-next": "",
                panel: l.next,
                calendar: M.value.next
              }, null, 8, ["panel", "calendar"]),
              C("div", aa, [
                ve(ee(Qe, {
                  months: N.value,
                  onUpdateMonth: M.value.next.setMonth
                }, null, 8, ["months", "onUpdateMonth"]), [
                  [pe, l.next.month]
                ]),
                ve(ee(qe, {
                  "as-prev-or-next": "",
                  years: M.value.next.years(),
                  onUpdateYear: M.value.next.setYear
                }, null, 8, ["years", "onUpdateYear"]), [
                  [pe, l.next.year]
                ]),
                ve(C("div", null, [
                  ee(Xe, { weeks: R.value }, null, 8, ["weeks"]),
                  ee(Je, {
                    "as-prev-or-next": "",
                    calendar: M.value.next,
                    weeks: R.value,
                    "as-range": y(),
                    "week-number": p.weekNumber,
                    onUpdateDate: f[4] || (f[4] = (v) => F(v))
                  }, null, 8, ["calendar", "weeks", "as-range", "week-number"])
                ], 512), [
                  [pe, l.next.calendar]
                ])
              ])
            ])) : ye("", !0)
          ])
        ]),
        e.autoApply ? ye("", !0) : (W(), Z("div", oa, [
          C("div", sa, [
            C("div", la, [
              C("button", {
                type: "button",
                class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                disabled: e.asSingle ? V.value.length < 1 : V.value.length < 2,
                onClick: f[5] || (f[5] = (v) => G()),
                textContent: ae(e.options.footer.apply)
              }, null, 8, ua)
            ])
          ])
        ]))
      ])
    ])) : ye("", !0) : (W(), Ue(le(Pt), {
      key: 0,
      id: "vtd",
      as: "div",
      class: "relative w-full"
    }, {
      default: Fe(({ open: v }) => [
        e.overlay && !e.disabled ? (W(), Ue(le(Or), {
          key: 0,
          class: "fixed inset-0 bg-black opacity-30"
        })) : ye("", !0),
        ee(le(Dr), {
          as: "label",
          class: "relative block"
        }, {
          default: Fe(() => [
            dt(p.$slots, "default", {
              value: k.value,
              placeholder: O.value,
              clear: S
            }, () => [
              ve(C("input", tr({
                ref_key: "VtdInputRef",
                ref: I
              }, p.$attrs, {
                "onUpdate:modelValue": f[0] || (f[0] = (i) => k.value = i),
                type: "text",
                class: ["relative block w-full", [
                  e.disabled ? "cursor-default opacity-50" : "opacity-100",
                  p.inputClasses || "pl-3 pr-12 py-2.5 rounded-lg overflow-hidden border-solid text-sm text-vtd-secondary-700 placeholder-vtd-secondary-400 transition-colors bg-white border border-vtd-secondary-300 focus:border-vtd-primary-300 focus:ring focus:ring-vtd-primary-500 focus:ring-opacity-10 focus:outline-none dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700 dark:text-vtd-secondary-100 dark:placeholder-vtd-secondary-500 dark:focus:border-vtd-primary-500 dark:focus:ring-opacity-20"
                ]],
                disabled: e.disabled,
                autocomplete: "off",
                "data-lpignore": "true",
                "data-form-type": "other",
                placeholder: O.value,
                onKeyup: Se(B, ["stop"]),
                onKeydown: f[1] || (f[1] = Se(() => {
                }, ["stop"]))
              }), null, 16, _n), [
                [rr, k.value]
              ]),
              C("div", Dn, [
                C("button", {
                  type: "button",
                  disabled: e.disabled,
                  class: Ye([
                    e.disabled ? "cursor-default opacity-50" : "opacity-100",
                    "px-2 py-1 mr-1 focus:outline-none text-vtd-secondary-400 dark:text-opacity-70 rounded-md"
                  ]),
                  onClick: f[2] || (f[2] = (i) => {
                    var $;
                    return e.disabled ? !1 : k.value ? S() : ($ = I.value) == null ? void 0 : $.focus();
                  })
                }, [
                  dt(p.$slots, "inputIcon", { value: k.value }, () => [
                    (W(), Z("svg", Pn, [
                      k.value ? (W(), Z("path", Vn)) : (W(), Z("path", Tn))
                    ]))
                  ])
                ], 10, On)
              ])
            ])
          ]),
          _: 3
        }),
        ee(mt, {
          "enter-from-class": "opacity-0 translate-y-3",
          "enter-to-class": "opacity-100 translate-y-0",
          "enter-active-class": "transform transition ease-out duration-200",
          "leave-active-class": "transform transition ease-in duration-150",
          "leave-from-class": "opacity-100 translate-y-0",
          "leave-to-class": "opacity-0 translate-y-3"
        }, {
          default: Fe(() => [
            e.disabled ? ye("", !0) : (W(), Ue(le(Pr), {
              key: 0,
              as: "div",
              class: "relative z-50"
            }, {
              default: Fe(({ close: i }) => [
                C("div", {
                  class: Ye(["absolute z-50 top-full sm:mt-2.5", Ie(v)])
                }, [
                  C("div", {
                    ref_key: "VtdRef",
                    ref: T,
                    class: "fixed inset-0 z-50 overflow-y-auto sm:overflow-visible sm:static sm:z-auto bg-white dark:bg-vtd-secondary-800 sm:rounded-lg shadow-sm"
                  }, [
                    C("div", {
                      class: Ye(["vtd-datepicker static sm:relative w-full bg-white sm:rounded-lg sm:shadow-sm border-0 sm:border border-black/[.1] px-3 py-3 sm:px-4 sm:py-4 dark:bg-vtd-secondary-800 dark:border-vtd-secondary-700/[1]", Re(v)])
                    }, [
                      C("div", An, [
                        e.shortcuts ? (W(), Ue(vt, {
                          key: 0,
                          shortcuts: e.shortcuts,
                          "as-range": y(),
                          "as-single": e.asSingle,
                          i18n: e.options.shortcuts,
                          close: i
                        }, null, 8, ["shortcuts", "as-range", "as-single", "i18n", "close"])) : ye("", !0),
                        C("div", Yn, [
                          y() && !e.asSingle ? (W(), Z("div", Cn, En)) : ye("", !0),
                          C("div", {
                            class: Ye(["relative", {
                              "mb-3 sm:mb-0 sm:mr-2 w-full md:w-1/2 lg:w-80": y() && !e.asSingle,
                              "w-full": !y() && e.asSingle
                            }])
                          }, [
                            ee(Ze, {
                              panel: l.previous,
                              calendar: M.value.previous
                            }, null, 8, ["panel", "calendar"]),
                            C("div", Ln, [
                              ve(ee(Qe, {
                                months: N.value,
                                onUpdateMonth: M.value.previous.setMonth
                              }, null, 8, ["months", "onUpdateMonth"]), [
                                [pe, l.previous.month]
                              ]),
                              ve(ee(qe, {
                                years: M.value.previous.years(),
                                onUpdateYear: M.value.previous.setYear
                              }, null, 8, ["years", "onUpdateYear"]), [
                                [pe, l.previous.year]
                              ]),
                              ve(C("div", null, [
                                ee(Xe, { weeks: R.value }, null, 8, ["weeks"]),
                                ee(Je, {
                                  calendar: M.value.previous,
                                  weeks: R.value,
                                  "as-range": y(),
                                  "week-number": p.weekNumber,
                                  onUpdateDate: ($) => F($, i)
                                }, null, 8, ["calendar", "weeks", "as-range", "week-number", "onUpdateDate"])
                              ], 512), [
                                [pe, l.previous.calendar]
                              ])
                            ])
                          ], 2),
                          y() && !e.asSingle ? (W(), Z("div", Fn, [
                            ee(Ze, {
                              "as-prev-or-next": "",
                              panel: l.next,
                              calendar: M.value.next
                            }, null, 8, ["panel", "calendar"]),
                            C("div", Bn, [
                              ve(ee(Qe, {
                                months: N.value,
                                onUpdateMonth: M.value.next.setMonth
                              }, null, 8, ["months", "onUpdateMonth"]), [
                                [pe, l.next.month]
                              ]),
                              ve(ee(qe, {
                                "as-prev-or-next": "",
                                years: M.value.next.years(),
                                onUpdateYear: M.value.next.setYear
                              }, null, 8, ["years", "onUpdateYear"]), [
                                [pe, l.next.year]
                              ]),
                              ve(C("div", null, [
                                ee(Xe, { weeks: R.value }, null, 8, ["weeks"]),
                                ee(Je, {
                                  "as-prev-or-next": "",
                                  calendar: M.value.next,
                                  weeks: R.value,
                                  "as-range": y(),
                                  "week-number": p.weekNumber,
                                  onUpdateDate: ($) => F($, i)
                                }, null, 8, ["calendar", "weeks", "as-range", "week-number", "onUpdateDate"])
                              ], 512), [
                                [pe, l.next.calendar]
                              ])
                            ])
                          ])) : ye("", !0)
                        ])
                      ]),
                      e.autoApply ? (W(), Z("div", Wn, [
                        C("div", Kn, [
                          C("div", zn, [
                            C("button", {
                              type: "button",
                              class: "away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              onClick: ($) => i(),
                              textContent: ae(e.options.footer.cancel)
                            }, null, 8, Gn)
                          ])
                        ])
                      ])) : (W(), Z("div", Nn, [
                        C("div", Rn, [
                          C("div", In, [
                            C("button", {
                              type: "button",
                              class: "away-apply-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-vtd-primary-600 text-base font-medium text-white hover:bg-vtd-primary-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800 disabled:cursor-not-allowed",
                              disabled: e.asSingle ? V.value.length < 1 : V.value.length < 2,
                              onClick: ($) => G(i),
                              textContent: ae(e.options.footer.apply)
                            }, null, 8, Hn),
                            C("button", {
                              type: "button",
                              class: "mt-3 away-cancel-picker w-full transition ease-out duration-300 inline-flex justify-center rounded-md border border-vtd-secondary-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-vtd-secondary-700 hover:bg-vtd-secondary-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-vtd-primary-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm dark:ring-offset-vtd-secondary-800",
                              onClick: ($) => i(),
                              textContent: ae(e.options.footer.cancel)
                            }, null, 8, Un)
                          ])
                        ])
                      ]))
                    ], 2)
                  ], 512)
                ], 2)
              ]),
              _: 2
            }, 1024))
          ]),
          _: 2
        }, 1024)
      ]),
      _: 3
    }));
  }
});
const Gt = /* @__PURE__ */ (() => {
  const t = ia;
  return t.install = (n) => {
    n.component("VueTailwindDatepicker", t);
  }, t;
})(), da = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Gt
}, Symbol.toStringTag, { value: "Module" }));
Object.entries(da).forEach(([t, n]) => {
  t !== "default" && (Gt[t] = n);
});
export {
  Gt as default
};
